---
title: "Figure 7 Infantis"
author: "Joao Carlos Gomes-Neto"
date: "3/3/2021"
output: html_document
---

Load packages

```{r, include = FALSE}
# Load packages

library(tidyverse)
library(skimr)
library(RColorBrewer)
library(gridExtra)
library(lattice)
library(ggpubr)
library(vegan)
library(reshape2)
library(ggrepel)
library(ggnewscale)
library(forcats)
library(naniar)
library(data.table)
library(rARPACK)
library(logisticPCA)
library(cluster)
library(factoextra)
library(NbClust)
library(plotly)
library(visdat)
library(ggtree)
library(Rtsne)
library(ggridges)
```


Salmonella Infantis phylogeny

```{r, include = FALSE}
# enter metadata

d1 <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/input_data/metadata_cleaned_final_new.csv')

# rename variables

d2 <- d1 %>% select(id, collection_date, source)

# Combine all years less than 2015 into one group

d2 <- d2 %>%  mutate_all(na_if, "missing")

# change NA in the year column to zero

d2$collection_date[is.na(d2$collection_date)] <- 0

# transform year to numeric

d2 <- d2 %>% mutate(collection_date = as.numeric(collection_date))

# create a new column called Year

d2 <- mutate(d2, year = ifelse(collection_date == 2016, "2016",
                                    ifelse(collection_date == 2017, "2017",
                                            ifelse(collection_date == 2018, "2018",
                                                      ifelse(collection_date == 2019, "2019",
                                                             "<= 2015"
                                           )))))

# check for missing values 

vis_miss(d2)

# filter only needed columns 

d3b <- d2 %>% select(id, year, source)
# Enter the phylogeny for Typhimurium group1

tree <- read.tree("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/input_data/infantis_phylogeny.tree")

#############################################

# BAPS1

baps <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/input_data/fastbaps_partition_baps_prior_l6.csv')

# Changing column names 

colnames(baps)[1:2] <- c("id", "baps_1") 

# Select colunms id and baps_1
b1 <- baps %>% 
              select(id, baps_1)


d1 <- b1

#############################################################################################

# ST

# enter mlst data 
mlst <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/input_data/salmonellast_output.csv')

# QC the data

skim(mlst)

# Generate the id column 

mlst$id2 <- sapply(strsplit(as.character(mlst$FILE),'_'), "[", 1)

# Select columns of interest

s1 <- mlst %>%
          select(id2, ST) %>%
            mutate_all(na_if, "-") %>%
              mutate_all(na_if, "?") %>%
                  mutate(id = id2)

# st

s1$st <- ifelse(s1$ST == 32, "ST32", "Other STs")

d2 <- s1

d2 <- d2 %>% select(-ST)

#############################################################################################

# CGMLST

# Enter SISTR results

sistr <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/input_data/sistr_all.csv')

# QC the data 

skim(sistr)

# Generate the id column 

sistr$id <- sapply(strsplit(as.character(sistr$genome),'_'), "[", 1)

# Select columns 

sis1 <- sistr %>%
            select(id, cgmlst_ST, serovar_cgmlst) %>%
            mutate_all(na_if, "-") %>%
              mutate_all(na_if, "?") 

sis1$serovar <- ifelse(sis1$serovar_cgmlst == "Infantis", "Infantis",
                            "Other serovars")

sis1 <- mutate(sis1, cgmlst = ifelse(cgmlst_ST %in% 2242423463, "cgMLST 2242423463",
                                            "Other cgMLSTs"))

d3 <- sis1

d3 <- d3 %>% select(id, cgmlst)

#############################################################################################

# KMER CLUSTER 

# Enter kmer-matrix

data1 <- read_csv("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/kmer_matrix/kmer_infantis.csv")

# generate cluster membership

#  Non-supervised kmeans clustering analysis

cluster_data <- data1 %>% select(tSNE1, tSNE2)

# combining both approaches we determine that 2 clusters are enough 
set.seed(123)
km.final <- kmeans(cluster_data, 2)
data1$kmer_cluster <- km.final$cluster
data1$kmer_cluster <- as.factor(data1$kmer_cluster)

# select the column we need

d4 <- data1 %>% select(id, kmer_cluster)

#############################################################################################

# SNP CLUSTER

# enter snp matrix

data2 <- read_csv("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/snp_matrix/snp_infantis.csv")

#  Non-supervised kmeans clustering analysis

cluster_data <- data2 %>% select(tSNE1, tSNE2)

# combining both approaches we determine that 2 clusters are enough 
set.seed(123)
km.final <- kmeans(cluster_data, 3)
data2$snp_cluster <- km.final$cluster
data2$snp_cluster <- as.factor(data2$snp_cluster)

# filter the column

d5 <- data2 %>% select(id, snp_cluster)

#############################################################################################

# PCA CLUSTER

data1 <- read_csv("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/pca_infantis/pca_data2.csv")
data2 <- read_csv("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/pca_infantis/cluster2.csv")

d6 <- data1 %>% select(-X1)
d7 <- data2 %>% select(-X1)
#############################################################################################
#############################################################################################

# combining all datasets

d8 <- left_join(d1, d2, on = "id")
d9 <- left_join(d8, d3, on = "id")
d10 <- left_join(d9, d4, on = "id")
d11 <- left_join(d10, d5, on = "id")
d12 <- left_join(d11, d6, on = "id")
d13 <- left_join(d12, d7, on = "id")

# if NA for ST call others 

d14 <- d13 %>% mutate(st = replace_na(st, "Other STs"))
d15 <- d14 %>% mutate(pca_cluster = cluster) %>% select(-cluster)
d15 <- d15 %>% mutate(baps_1 = as.factor(baps_1))
d15 <- d15 %>% mutate(pca_cluster = as.factor(pca_cluster))


# drop columns

d16 <- d15 %>% select(id, baps_1, st, cgmlst, pca_cluster)

# create a new colum

d17 <- d16 %>% mutate(pca = ifelse(pca_cluster == "1", "Cluster 1", "Cluster 2"))
d17 <- d17 %>% select(-pca_cluster)


# merge data

d18 <- left_join(d17, d3b, on = "id")
########################################################################

# enter the cdc metadata

#d19 <- read_csv("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/input_data/infantis_downloaded_sra_ids.txt", col_names = FALSE)

#d19 <- rename(d19, id = X1)

# create a new column with human as source

#d19$cdc <- "Human-CDC"

# merge datasets

#d20 <- left_join(d18, d19, on = "id")
########################################################################
########################################################################

# Make id the rowname or index

d18e <- column_to_rownames(d18, var = "id") 

# Create the tree

i1 <- ggtree(tree, layout = "circular") + xlim(-50, NA)


f1 <- gheatmap(i1, d18e, offset=.0, width=10, colnames = FALSE) +
  scale_fill_manual(values = c("darkblue", "gray", "coral", 
                               "darkred", "darkblue", 
                                 "black", "orange",
                               "darkblue", "gray",
                               "darkgreen", "red", "darkblue", "darkred", "orange",
                                "gray", "darkgreen", "darkblue", "coral", "purple",
                               "darkgreen", "gray"
                               ),
                    breaks = c(
                               "1", "2", "3",
                               "ST32", "Other STs",
                               "cgMLST 2242423463", "Other cgMLSTs",
                               "Cluster 1", "Cluster 2",
                               "<= 2015", "2016", "2017", "2018", "2019",
                               "Bovine", "Human", "Poultry", "Swine", "Environmental/Others",
                               "Human-CDC", "Others-CDC"
                               ),
                    name="") +
    ggtitle("") +
  theme(plot.title = element_text(size = 50, face = "bold"), 
        legend.title=element_text(size=26, face = "bold"), 
    legend.text=element_text(size=22))
f1
```


Plot the temporal changes in the clusters over time

```{r, include = FALSE}
# enter metadata

d1 <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/input_data/metadata_cleaned_final_new.csv')

# rename variables

d2 <- d1 %>% select(id, collection_date, source)

# Combine all years less than 2015 into one group

d2 <- d2 %>%  mutate_all(na_if, "missing")

# change NA in the year column to zero

d2$collection_date[is.na(d2$collection_date)] <- 0

# transform year to numeric

d2 <- d2 %>% mutate(collection_date = as.numeric(collection_date))

# create a new column called Year

d2 <- mutate(d2, year = ifelse(collection_date == 2007, "2007",
                               ifelse(collection_date == 2008, "2008",
                                      ifelse(collection_date == 2009, "2009",
                                             ifelse(collection_date == 2010, "2010",
                                                    ifelse(collection_date == 2011, "2011",
                                                           ifelse(collection_date == 2012, "2012",
                                                                  ifelse(collection_date == 2013, "2013",
                                                                         ifelse(collection_date == 2014, "2014",
                                                                               ifelse(collection_date == 2015, "2015",
                                                                                 ifelse(collection_date == 2016, "2016",
                                    ifelse(collection_date == 2017, "2017",
                                            ifelse(collection_date == 2018, "2018",
                                                      ifelse(collection_date == 2019, "2019",
                                                             "<= 2006"
                                           ))))))))))))))

# check for missing values 

vis_miss(d2)

# filter only needed columns 

d3b <- d2 %>% select(id, year, source)
#############################################

# BAPS1

baps <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/input_data/fastbaps_partition_baps_prior_l6.csv')

# Changing column names 

colnames(baps)[1:2] <- c("id", "baps_1") 

# Select colunms id and baps_1
b1 <- baps %>% 
              select(id, baps_1)


d1 <- b1

#############################################################################################

# ST

# enter mlst data 
mlst <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/input_data/salmonellast_output.csv')

# QC the data

skim(mlst)

# Generate the id column 

mlst$id2 <- sapply(strsplit(as.character(mlst$FILE),'_'), "[", 1)

# Select columns of interest

s1 <- mlst %>%
          select(id2, ST) %>%
            mutate_all(na_if, "-") %>%
              mutate_all(na_if, "?") %>%
                  mutate(id = id2)

# st

s1$st <- ifelse(s1$ST == 32, "ST32", "Other STs")

d2 <- s1

d2 <- d2 %>% select(-ST)

#############################################################################################

# CGMLST

# Enter SISTR results

sistr <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/input_data/sistr_all.csv')

# QC the data 

skim(sistr)

# Generate the id column 

sistr$id <- sapply(strsplit(as.character(sistr$genome),'_'), "[", 1)

# Select columns 

sis1 <- sistr %>%
            select(id, cgmlst_ST, serovar_cgmlst) %>%
            mutate_all(na_if, "-") %>%
              mutate_all(na_if, "?") 

sis1$serovar <- ifelse(sis1$serovar_cgmlst == "Infantis", "Infantis",
                            "Other serovars")

sis1 <- mutate(sis1, cgmlst = ifelse(cgmlst_ST %in% 2242423463, "cgMLST 2242423463",
                                            "Other cgMLSTs"))

d3 <- sis1

d3 <- d3 %>% select(id, cgmlst)

#############################################################################################

# KMER CLUSTER 

# Enter kmer-matrix

data1 <- read_csv("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/kmer_matrix/kmer_infantis.csv")

# generate cluster membership

#  Non-supervised kmeans clustering analysis

cluster_data <- data1 %>% select(tSNE1, tSNE2)

# combining both approaches we determine that 2 clusters are enough 
set.seed(123)
km.final <- kmeans(cluster_data, 2)
data1$kmer_cluster <- km.final$cluster
data1$kmer_cluster <- as.factor(data1$kmer_cluster)

# select the column we need

d4 <- data1 %>% select(id, kmer_cluster)

#############################################################################################

# SNP CLUSTER

# enter snp matrix

data2 <- read_csv("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/snp_matrix/snp_infantis.csv")

#  Non-supervised kmeans clustering analysis

cluster_data <- data2 %>% select(tSNE1, tSNE2)

# combining both approaches we determine that 2 clusters are enough 
set.seed(123)
km.final <- kmeans(cluster_data, 3)
data2$snp_cluster <- km.final$cluster
data2$snp_cluster <- as.factor(data2$snp_cluster)

# filter the column

d5 <- data2 %>% select(id, snp_cluster)

#############################################################################################

# PCA CLUSTER

data1 <- read_csv("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/pca_infantis/pca_data2.csv")
data2 <- read_csv("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/pca_infantis/cluster2.csv")

d6 <- data1 %>% select(-X1)
d7 <- data2 %>% select(-X1)
#############################################################################################
#############################################################################################

# combining all datasets

d8 <- left_join(d1, d2, on = "id")
d9 <- left_join(d8, d3, on = "id")
d10 <- left_join(d9, d4, on = "id")
d11 <- left_join(d10, d5, on = "id")
d12 <- left_join(d11, d6, on = "id")
d13 <- left_join(d12, d7, on = "id")

# if NA for ST call others 

d14 <- d13 %>% mutate(st = replace_na(st, "Other STs"))
d15 <- d14 %>% mutate(pca_cluster = cluster) %>% select(-cluster)
d15 <- d15 %>% mutate(baps_1 = as.factor(baps_1))
d15 <- d15 %>% mutate(pca_cluster = as.factor(pca_cluster))


# drop columns

d16 <- d15 %>% select(id, baps_1, st, cgmlst, pca_cluster)

# create a new colum

d17 <- d16 %>% mutate(pca = ifelse(pca_cluster == "1", "Cluster 1", "Cluster 2"))
d17 <- d17 %>% select(-pca_cluster)


# merge data

d18 <- left_join(d17, d3b, on = "id")

data1 <- d18 %>%
              select(pca, year, source) %>%
              group_by(source, year, pca) %>%
              dplyr::summarize(n = n()) %>%
              mutate(total = sum(n)) %>% 
              mutate(prop = n/total*100) 

# plot data

plot1 <- ggplot(data1, aes(x = year, y = prop, color = pca, group = pca)) + ylim(0, 100) +
  theme_bw() +  xlab("Years") +
  ylab("Proportion") +
    theme(axis.text.x = element_text(angle = 80, hjust = 1, size = 20)) +
  theme(axis.title.x = element_text(size = 35, face = "bold")) +
  theme(axis.title.y = element_text(size = 35, face = "bold")) +
  theme(axis.text.y = element_text(angle = 0, hjust = 1, size = 30)) +
  theme(legend.title=element_text(size=35, face = "bold"))+
  theme(legend.text=element_text(size=30)) +
  scale_color_manual(name = "PCA-clusters", values=c("darkblue", "gray")) +
  theme(strip.text.x = element_text(size = 25, face = "bold")) +
  #guides(linetype = guide_legend(override.aes = list(size = 10)))  +
    geom_line(size=3) +
  theme(legend.key.width = unit(3,"cm")) +
  facet_wrap(~source) 
plot1
```

Plot PCA cluster with host information

```{r, include = FALSE}

#############################################

# BAPS1

baps <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/input_data/fastbaps_partition_baps_prior_l6.csv')

# Changing column names 

colnames(baps)[1:2] <- c("id", "baps_1") 

# Select colunms id and baps_1
b1 <- baps %>% 
              select(id, baps_1)


d1 <- b1

#############################################################################################

# ST

# enter mlst data 
mlst <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/input_data/salmonellast_output.csv')

# QC the data

skim(mlst)

# Generate the id column 

mlst$id2 <- sapply(strsplit(as.character(mlst$FILE),'_'), "[", 1)

# Select columns of interest

s1 <- mlst %>%
          select(id2, ST) %>%
            mutate_all(na_if, "-") %>%
              mutate_all(na_if, "?") %>%
                  mutate(id = id2)

# st

s1$st <- ifelse(s1$ST == 32, "ST32", "Other STs")

d2 <- s1

d2 <- d2 %>% select(-ST)

#############################################################################################

# CGMLST

# Enter SISTR results

sistr <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/input_data/sistr_all.csv')

# QC the data 

skim(sistr)

# Generate the id column 

sistr$id <- sapply(strsplit(as.character(sistr$genome),'_'), "[", 1)

# Select columns 

sis1 <- sistr %>%
            select(id, cgmlst_ST, serovar_cgmlst) %>%
            mutate_all(na_if, "-") %>%
              mutate_all(na_if, "?") 

sis1$serovar <- ifelse(sis1$serovar_cgmlst == "Infantis", "Infantis",
                            "Other serovars")

sis1 <- mutate(sis1, cgmlst = ifelse(cgmlst_ST %in% 2242423463, "cgMLST 2242423463",
                                            "Other cgMLSTs"))

d3 <- sis1

d3 <- d3 %>% select(id, cgmlst)

#############################################################################################

# KMER CLUSTER 

# Enter kmer-matrix

data1 <- read_csv("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/kmer_matrix/kmer_infantis.csv")

# generate cluster membership

#  Non-supervised kmeans clustering analysis

cluster_data <- data1 %>% select(tSNE1, tSNE2)

# combining both approaches we determine that 2 clusters are enough 
set.seed(123)
km.final <- kmeans(cluster_data, 2)
data1$kmer_cluster <- km.final$cluster
data1$kmer_cluster <- as.factor(data1$kmer_cluster)

# select the column we need

d4 <- data1 %>% select(id, kmer_cluster)

#############################################################################################

# SNP CLUSTER

# enter snp matrix

data2 <- read_csv("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/snp_matrix/snp_infantis.csv")

#  Non-supervised kmeans clustering analysis

cluster_data <- data2 %>% select(tSNE1, tSNE2)

# combining both approaches we determine that 2 clusters are enough 
set.seed(123)
km.final <- kmeans(cluster_data, 3)
data2$snp_cluster <- km.final$cluster
data2$snp_cluster <- as.factor(data2$snp_cluster)

# filter the column

d5 <- data2 %>% select(id, snp_cluster)

#############################################################################################

# PCA CLUSTER

data1 <- read_csv("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/pca_infantis/pca_data2.csv")
data2 <- read_csv("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/pca_infantis/cluster2.csv")

d6 <- data1 %>% select(-X1)
d7 <- data2 %>% select(-X1)
#############################################################################################
#############################################################################################

# combining all datasets

d8 <- left_join(d1, d2, on = "id")
d9 <- left_join(d8, d3, on = "id")
d10 <- left_join(d9, d4, on = "id")
d11 <- left_join(d10, d5, on = "id")
d12 <- left_join(d11, d6, on = "id")
d13 <- left_join(d12, d7, on = "id")

# if NA for ST call others 

d14 <- d13 %>% mutate(st = replace_na(st, "Other STs"))
d15 <- d14 %>% mutate(pca_cluster = cluster) %>% select(-cluster)
d15 <- d15 %>% mutate(baps_1 = as.factor(baps_1))
d15 <- d15 %>% mutate(pca_cluster = as.factor(pca_cluster))

d16 <- left_join(d15, d3b, on = "id")
d16$source <- factor(d16$source, levels=c("Bovine", "Human", "Poultry", "Swine", "Environmental/Others"))

p6 <- ggplot(d16, aes(x = PC1, y = PC2, color = source)) + 
  theme_bw() + 
  xlab("PC1") +
  ylab("PC2") +
  theme(axis.text.y = element_text(size = 40)) +
  theme(axis.title.y = element_text(size = 45, face = "bold")) +
  theme(axis.title.x = element_text(size = 45, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 1, size = 40)) +
  scale_color_manual(values=c("gray", "darkgreen", "darkblue", "coral", "purple")) +
  labs(color='Source') + 
  geom_jitter(pch = 19, height = 1, width = 1, size = 4) +
  theme(legend.title=element_text(size=50, face = "bold")) +
  theme(legend.text=element_text(size=40)) +
  guides(color = guide_legend(override.aes = list(size=20)))
p6

##########################################################

d16b <- d16 %>% select(pca_cluster, source)
d16b <- d16b %>% mutate(pca = ifelse(pca_cluster == "1", "Cluster 1", "Cluster 2"))

# calculate the proportion of pca_cluster by source

data1 <- d16b %>%
              select(source, pca) %>%
              group_by(source, pca) %>%
              dplyr::summarise(n = n()) %>%
              mutate(total = sum(n)) %>% 
              mutate(prop = n/total*100) 

# plot data
data1$pca <- factor(data1$pca, levels=c("Cluster 1", "Cluster 2"))

plot2 <- ggplot(data1, aes(x = prop, y = source, fill = pca)) + xlim(0, 101) +
  theme_bw() +  xlab("Proportion") +
  ylab("") +
    theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 30)) +
  theme(axis.title.x = element_text(size = 35, face = "bold")) +
  theme(axis.text.y = element_text(angle = 0, hjust = 1, size = 30)) +
  theme(strip.text.x = element_text(size = 30, face = "bold")) +
  theme(legend.title=element_text(size=35, face = "bold"))+
  theme(legend.text=element_text(size=30)) +
  scale_fill_manual(name = "PCA-clusters", values=c("darkblue", "gray")) +
    geom_bar(position="stack", stat="identity")
plot2

##################################################################

# Plot whole-genome distance using mash 

d17 <- d16 %>% select(id, pca_cluster, source)

d18 <- read_tsv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/assemblies_kmer/ref_infantis_table.csv', col_names = TRUE)

# merge files

d19 <- left_join(d17, d18, on = "id")
d19 <- d19 %>% mutate(pca = ifelse(pca_cluster == "1", "Cluster 1", "Cluster 2"))
d19$pca <- factor(d19$pca, levels=c("Cluster 1", "Cluster 2"))
d20 <- column_to_rownames(d19, var = "id")
d21 <- d20 %>% select(pca, distance)
d22 <- d21 %>% group_by(pca) %>% mutate(sdx = sd(distance))

# plot distance distribution

p1 <- ggplot(d19, aes(x = distance, y = pca, color = pca)) +
  geom_jitter(width = 0.01, size = 3) +
  xlim(-0.1, 0.5) +
  theme_minimal() + 
  theme(legend.position = "none") +
  xlab("Whole-genome Kmer-based distance") +
  ylab("") +
  theme(axis.text.y = element_text(size = 30)) +
  theme(axis.title.y = element_text(size = 35, face = "bold")) +
  theme(axis.title.x = element_text(size = 35, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 1, size = 30)) +
  theme(strip.text.x = element_text(size = 35)) +
  scale_color_manual(values=c("darkblue", "darkred")) 
p1

# calculate and plot the median and range of the kmer distance
data_summary <- function(data, varname, groupnames){
  require(plyr)
  summary_func <- function(x, col){
    c(median = median(x[[col]], na.rm=TRUE),
      mean = mean(x[[col]], na.rm=TRUE),
      sd = sd(x[[col]], na.rm=TRUE),
      range = range(x[[col]], na.rm=TRUE))
  }
  data_sum<-ddply(data, groupnames, .fun=summary_func,
                  varname)
  data_sum <- rename(data_sum, c("mean" = varname))
 return(data_sum)
}
df2 <- data_summary(d21, varname="distance", 
                    groupnames=c("pca"))

p2 <- ggplot(df2, aes(x = median, y = pca, color = pca)) + 
geom_pointrange(aes(xmin = median-range1, xmax = median+range2, size = 4)) +
  xlim(0, 0.3) +
  theme_classic() + 
  theme(legend.position = "none") +
  xlab("Whole-genome Kmer-based distance") +
  ylab("") +
  theme(axis.text.y = element_blank(),
  axis.ticks.y = element_blank()) +
  theme( axis.line = element_line(colour = "black", 
                      size = 1, linetype = "solid"))+
  #theme(axis.text.y = element_text(size = 40)) +
  theme(axis.title.y = element_text(size = 35, face = "bold")) +
  theme(axis.title.x = element_text(size = 45, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 1, size = 50)) +
  scale_color_manual(values=c("darkblue", "darkred")) 
p2

# plot by source

d20 <- column_to_rownames(d19, var = "id")
d21 <- d20 %>% select(source, distance)
# calculate and plot the median and range of the kmer distance
data_summary <- function(data, varname, groupnames){
  require(plyr)
  summary_func <- function(x, col){
    c(median = median(x[[col]], na.rm=TRUE),
      mean = mean(x[[col]], na.rm=TRUE),
      sd = sd(x[[col]], na.rm=TRUE),
      range = range(x[[col]], na.rm=TRUE))
  }
  data_sum<-ddply(data, groupnames, .fun=summary_func,
                  varname)
  data_sum <- rename(data_sum, c("mean" = varname))
 return(data_sum)
}
df3 <- data_summary(d21, varname="distance", 
                    groupnames=c("source"))
df3$source <- factor(df3$source, levels=c("Bovine", "Human", "Poultry", "Swine", "Environmental/Others"))


p3 <- ggplot(df3, aes(x = median, y = source, color = source)) + 
geom_pointrange(aes(xmin = median-range1, xmax = median+range2, size = 4)) +
  xlim(0, 0.31) +
  theme_classic() + 
  theme(legend.position = "none") +
  xlab("") +
  ylab("") +
  theme(axis.text.y = element_blank(),
  axis.ticks.y = element_blank()) +
   theme( axis.line = element_line(colour = "black", 
                      size = 1, linetype = "solid"))+
  xlab("Whole-genome Kmer-based distance") +
  #theme(axis.text.y = element_text(size = 40)) +
  #theme(axis.title.y = element_text(size = 35, face = "bold")) +
  theme(axis.title.x = element_text(size = 45, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 1, size = 50)) +
  scale_color_manual(values=c("gray", "darkgreen", "darkblue", "coral", "purple")) 
p3
```

Plot the temporal changes of the cgmlst over time

```{r, include = FALSE}

#############################################################################################

# CGMLST

# Enter SISTR results

sistr <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/input_data/sistr_all.csv')

# QC the data 

skim(sistr)

# Generate the id column 

sistr$id <- sapply(strsplit(as.character(sistr$genome),'_'), "[", 1)

# Select columns 

sis1 <- sistr %>%
            select(id, cgmlst_ST, serovar_cgmlst) %>%
            mutate_all(na_if, "-") %>%
              mutate_all(na_if, "?") 


sis1 <- mutate(sis1, cgmlst = ifelse(cgmlst_ST %in% 2242423463, "cgMLST 2242423463",
                                      ifelse(cgmlst_ST %in% 3873414194, "cgMLST 3873414194",
                                      ifelse(cgmlst_ST %in% 1267535865, "cgMLST 1267535865",
                                      ifelse(cgmlst_ST %in% 1582303201, "cgMLST 1582303201",
                                      ifelse(cgmlst_ST %in% 1999638313, "cgMLST 1999638313",
                                      ifelse(cgmlst_ST %in% 3087372233, "cgMLST 3087372233",
                                            "Other cgMLSTs")))))))

d1 <- sis1

d1 <- d1 %>% select(id, cgmlst)

#############################################################################################
#############################################################################################

# PCA CLUSTER

data2 <- read_csv("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/pca_infantis/cluster2.csv")

d2 <- data2 %>% select(-X1)
#############################################################################################
#############################################################################################

# combining all datasets

d3 <- left_join(d1, d2, on = "id")
d4 <- d3 %>% mutate(pca_cluster = cluster) %>% select(-cluster)
d4 <- d4 %>% mutate(pca_cluster = as.factor(pca_cluster))


# create a new column

d5 <- d4 %>% mutate(pca = ifelse(pca_cluster == "1", "Cluster 1", "Cluster 2"))
d6 <- d5 %>% select(-pca_cluster)


# merge data

d7 <- left_join(d6, d3b, on = "id")

data1 <- d7 %>%
              select(cgmlst, year, source) %>%
              group_by(source, year, cgmlst) %>%
              summarise(n = n()) %>%
              mutate(total = sum(n)) %>% 
              mutate(prop = n/total*100) 

data1$cgmlst <- factor(data1$cgmlst, levels=c("cgMLST 2242423463", "cgMLST 3873414194", 
                                              "cgMLST 1267535865", "cgMLST 1582303201",
                                              "cgMLST 1999638313", "cgMLST 3087372233",
                                              "Other cgMLSTs"))

# plot data

plot2 <- ggplot(data1, aes(x = year, y = prop, color = cgmlst, group = cgmlst)) + ylim(0, 100) +
  theme_bw() +  xlab("Years") +
  ylab("Proportion") +
    theme(axis.text.x = element_text(angle = 70, hjust = 1, size = 30)) +
  theme(axis.title.x = element_text(size = 35, face = "bold")) +
  theme(axis.title.y = element_text(size = 35, face = "bold")) +
  theme(axis.text.y = element_text(angle = 0, hjust = 1, size = 30)) +
  theme(legend.title=element_text(size=35, face = "bold"))+
  theme(legend.text=element_text(size=30)) +
  scale_color_manual(name = "cgMLST variants", values=c("black", "purple", "darkgreen", "gray", 
  "darkblue", "red", "orange")) +
  theme(strip.text.x = element_text(size = 20, face = "bold")) +
  #guides(linetype = guide_legend(override.aes = list(size = 10)))  +
    geom_line(size=3) +
  theme(legend.key.width = unit(3,"cm")) +
  facet_wrap(~source) 
plot2

# distribution of major cgmlsts across cluster

data2 <- d7 %>%
              select(pca, cgmlst) %>%
              drop_na() %>%
              group_by(pca, cgmlst) %>%
              summarise(n = n()) %>%
              mutate(total = sum(n)) %>% 
              mutate(prop = n/total*100) 

data2$cgmlst <- factor(data2$cgmlst, levels=c("cgMLST 2242423463", "cgMLST 3873414194", 
                                              "cgMLST 1267535865", "cgMLST 1582303201",
                                              "cgMLST 1999638313", "cgMLST 3087372233",
                                              "Other cgMLSTs"))
plot3 <- ggplot(data2, aes(x = prop, y = pca, fill = cgmlst)) + xlim(0, 101) +
  theme_bw() +  xlab("Proportion") +
  ylab("") +
    theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 35)) +
  theme(axis.title.x = element_text(size = 40, face = "bold")) +
  theme(axis.text.y = element_text(angle = 0, hjust = 1, size = 40)) +
  theme(strip.text.x = element_text(size = 30, face = "bold")) +
  theme(legend.title=element_text(size=35, face = "bold"))+
  theme(legend.text=element_text(size=30)) +
  scale_fill_manual(name = "cgMLST variants", values=c("black", "purple", "darkgreen", "gray", 
  "darkblue", "red", "orange")) +
    geom_bar(position="stack", stat="identity")
plot3
```


Simpson index of diversity of clusters based on cgMLST and baps2-6

```{r, include = FALSE}
#############################################

# BAPS1

baps <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/input_data/fastbaps_partition_baps_prior_l6.csv')

# Changing column names 

colnames(baps)[1:7] <- c("id", "BAPS1", "BAPS2", "BAPS3", "BAPS4", "BAPS5", "BAPS6") 


d1 <- baps

#############################################################################################
#############################################################################################

# CGMLST

# Enter SISTR results

sistr <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/input_data/sistr_all.csv')

# QC the data 

skim(sistr)

# Generate the id column 

sistr$id <- sapply(strsplit(as.character(sistr$genome),'_'), "[", 1)

# Select columns 

sis1 <- sistr %>%
            select(id, cgmlst_ST, serovar_cgmlst) %>%
            mutate_all(na_if, "-") %>%
              mutate_all(na_if, "?") 


d3 <- sis1

# PCA CLUSTER

data2 <- read_csv("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/pca_infantis/cluster2.csv")

d7 <- data2 %>% select(-X1)
#############################################################################################
#############################################################################################

# combining all datasets

d8 <- left_join(d1, d3, on = "id")
d9 <- left_join(d8, d7, on = "id")


# transform baps to factors

d9 <- d9 %>% mutate(BAPS2 = as.factor(BAPS2))
d9 <- d9 %>% mutate(BAPS3 = as.factor(BAPS3))
d9 <- d9 %>% mutate(BAPS4 = as.factor(BAPS4))
d9 <- d9 %>% mutate(BAPS5 = as.factor(BAPS5))
d9 <- d9 %>% mutate(BAPS6 = as.factor(BAPS6))

d10 <- d9 %>% select(-BAPS1)

# create a new colum

d10 <- d10 %>% mutate(pca = ifelse(cluster == "1", "Cluster 1", "Cluster 2"))

# cgMLST diversity 

cgmlst_div <- d10 %>%
              drop_na(cgmlst_ST) %>%
              mutate(cgmlst_ST = as.factor(cgmlst_ST)) %>%
              select(pca, cgmlst_ST) %>%
              group_by(pca, cgmlst_ST) %>%
              summarise(n = n()) %>%
              mutate(simpson = diversity(n, "simpson")) %>%
              group_by(pca) %>%
              summarise(simpson = mean(simpson)) %>%
              melt(id.vars=c("pca"), measure.vars="simpson",
                    variable.name="index", value.name="value") %>%
              mutate(strat = "cgMLST")

#####################################

# Baps based diversity 


# Baps2
baps2 <- d10 %>%
              select(pca, BAPS2) %>%
              drop_na(pca, BAPS2) %>%
              group_by(pca, BAPS2) %>%
              summarise(n = n()) %>%
              mutate(simpson = diversity(n, "simpson")) %>%
              group_by(pca) %>%
              summarise(simpson = mean(simpson)) %>%
              melt(id.vars=c("pca"), measure.vars="simpson",
                    variable.name="index", value.name="value") %>%
              mutate(strat = "BAPS2")


# Baps3
baps3 <- d10 %>%
              select(pca, BAPS3) %>%
              drop_na(pca, BAPS3) %>%
              group_by(pca, BAPS3) %>%
              summarise(n = n()) %>%
              mutate(simpson = diversity(n, "simpson")) %>%
              group_by(pca) %>%
              summarise(simpson = mean(simpson)) %>%
              melt(id.vars=c("pca"), measure.vars="simpson",
                    variable.name="index", value.name="value") %>%
              mutate(strat = "BAPS3")


# Baps4
baps4 <- d10 %>%
              select(pca, BAPS4) %>%
              drop_na(pca, BAPS4) %>%
              group_by(pca, BAPS4) %>%
              summarise(n = n()) %>%
              mutate(simpson = diversity(n, "simpson")) %>%
              group_by(pca) %>%
              summarise(simpson = mean(simpson)) %>%
              melt(id.vars=c("pca"), measure.vars="simpson",
                    variable.name="index", value.name="value") %>%
              mutate(strat = "BAPS4")

# Baps5
baps5 <- d10 %>%
              select(pca, BAPS5) %>%
              drop_na(pca, BAPS5) %>%
              group_by(pca, BAPS5) %>%
              summarise(n = n()) %>%
              mutate(simpson = diversity(n, "simpson")) %>%
              group_by(pca) %>%
              summarise(simpson = mean(simpson)) %>%
              melt(id.vars=c("pca"), measure.vars="simpson",
                    variable.name="index", value.name="value") %>%
              mutate(strat = "BAPS5")

# Baps6
baps6 <- d10 %>%
              select(pca, BAPS6) %>%
              drop_na(pca, BAPS6) %>%
              group_by(pca, BAPS6) %>%
              summarise(n = n()) %>%
              mutate(simpson = diversity(n, "simpson")) %>%
              group_by(pca) %>%
              summarise(simpson = mean(simpson)) %>%
              melt(id.vars=c("pca"), measure.vars="simpson",
                    variable.name="index", value.name="value") %>%
              mutate(strat = "BAPS6")

# Concatenate all datasets

d11 <- rbind(cgmlst_div, baps2, baps3, baps4, baps5, baps6)
d11$strat <- factor(d11$strat, levels=c("cgMLST", "BAPS2", "BAPS3",
                                              "BAPS4", "BAPS5", "BAPS6"))

# Plot Simpson's diversity analysis

fig7e <- ggplot(d11, aes(x = strat, y = value, group = pca, fill = pca)) + ylim(0,1.06)+
  theme_bw() + 
  xlab("") +
  ylab("Index value") +
  theme(axis.text.y = element_text(size = 45)) +
  theme(axis.title.y = element_text(size = 50, face = "bold")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 45)) +
  theme(axis.title.x = element_text(size = 26, face = "bold")) +
  theme(strip.text.x = element_text(size = 36))+
  theme(legend.title=element_text(size=35, face = "bold"))+
  theme(legend.text=element_text(size=30)) +
   geom_bar(position="dodge", stat="identity") +
  scale_fill_manual(name = "PCA-clusters", values=c("darkblue", "gray")) 
fig7e
```

Assess the clonality of each PCA cluster using BAPS1-6 haplotype analysis 

```{r, include = FASE}
# BAPS1

baps <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/input_data/fastbaps_partition_baps_prior_l6.csv')

# Changing column names 

colnames(baps)[1:7] <- c("id", "BAPS1", "BAPS2", "BAPS3", "BAPS4", "BAPS5", "BAPS6") 


d1 <- baps

# PCA CLUSTER

data2 <- read_csv("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/pca_infantis/cluster2.csv")

d2 <- data2 %>% select(-X1)
#############################################################################################
#############################################################################################

# combining all datasets

d3 <- left_join(d1, d2, on = "id")
d3 <- d3 %>% mutate(pca = ifelse(cluster == "1", "Cluster 1", "Cluster 2"))
d4 <- d3 %>% select(-cluster)

#####################################

# Baps based diversity 

# Baps1
baps1 <- d4 %>%
              select(pca, BAPS1) %>%
              drop_na(pca, BAPS1) %>%
              group_by(pca, BAPS1) %>%
              summarise(n = n()) %>%
              mutate(total = sum(n)) %>% 
              mutate(prop = n/total*100) %>%
              mutate(group = "BAPS1") %>%
              rename(baps = BAPS1)

# Baps2
baps2 <- d4 %>%
              select(pca, BAPS2) %>%
              drop_na(pca, BAPS2) %>%
              group_by(pca, BAPS2) %>%
              summarise(n = n()) %>%
              mutate(total = sum(n)) %>% 
              mutate(prop = n/total*100) %>%
              mutate(group = "BAPS2") %>%
              rename(baps = BAPS2)

# Baps3
baps3 <- d4 %>%
              select(pca, BAPS3) %>%
              drop_na(pca, BAPS3) %>%
              group_by(pca, BAPS3) %>%
              summarise(n = n()) %>%
              mutate(total = sum(n)) %>% 
              mutate(prop = n/total*100) %>%
              mutate(group = "BAPS3") %>%
              rename(baps = BAPS3)


# Baps4
baps4 <- d4 %>%
              select(pca, BAPS4) %>%
              drop_na(pca, BAPS4) %>%
              group_by(pca, BAPS4) %>%
              summarise(n = n()) %>%
              mutate(total = sum(n)) %>% 
              mutate(prop = n/total*100) %>%
              mutate(group = "BAPS4") %>%
              rename(baps = BAPS4)

# Baps5
baps5 <- d4 %>%
              select(pca, BAPS5) %>%
              drop_na(pca, BAPS5) %>%
              group_by(pca, BAPS5) %>%
              summarise(n = n()) %>%
              mutate(total = sum(n)) %>% 
              mutate(prop = n/total*100) %>%
              mutate(group = "BAPS5") %>%
              rename(baps = BAPS5)

              # Baps6
baps6 <- d4 %>%
              select(pca, BAPS6) %>%
              drop_na(pca, BAPS6) %>%
              group_by(pca, BAPS6) %>%
              summarise(n = n()) %>%
              mutate(total = sum(n)) %>% 
              mutate(prop = n/total*100) %>%
              mutate(group = "BAPS6") %>%
              rename(baps = BAPS6)

# Concatenate all datasets

d5 <- rbind(baps1, baps2, baps3, baps4, baps5, baps6)
d5$group <- factor(d5$group, levels=c("BAPS1", "BAPS2", "BAPS3",
                                              "BAPS4", "BAPS5", "BAPS6"))

# plot data

plot1 <- ggplot(d5, aes(x = baps, y = prop, fill = pca))  + xlim(0, 100) +
  theme_bw() +
  geom_col(position = "dodge")+
  xlab("BAPS sub-groups (haplotypes)") +
  ylab("Proportion") +
    theme(axis.text.x = element_text(angle = 0, size = 25)) +
  theme(axis.title.x = element_text(size = 35, face = "bold")) +
  theme(axis.title.y = element_text(size = 35, face = "bold")) +
  theme(axis.text.y = element_text(angle = 0, hjust = 1, size = 25)) +
  theme(legend.title=element_text(size=35, face = "bold"))+
  theme(legend.text=element_text(size=25)) +
  scale_fill_manual(name = "PCA-clusters", values=c("darkblue", "orange")) +
  theme(strip.text.x = element_text(size = 35)) +
  facet_wrap(~group, ncol = 3) 
plot1
```

Salmonella infantis - distribution of SNPs 

```{r, include = FALSE}

# enter the snp data

data1 <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/input_data/distance_snp_sites_infantis_columns.csv')

# filter out distance between itself

data2 <- data1 %>% filter_(~id_1 != id_2)

# distribution of snps 

summary(data2$dist)

# get cluster information

cluster_info <- d18 %>% select(pca)
cluster_info <- rownames_to_column(cluster_info, var = "id")

# select id_1

d1 <- data2 %>% select(-id_2) %>% rename(id = id_1)
d2 <- left_join(d1, cluster_info, on = "id")
d3 <- d2 %>% rename(id_1 = id)
d3 <- d3 %>% rename(pca_1 = pca) %>% select(-dist)

# select id_2

d1b <- data2 %>% select(-id_1) %>% rename(id = id_2)
d2b <- left_join(d1b, cluster_info, on = "id")
d3b <- d2b %>% rename(id_2 = id)
d3b <- d3b %>% rename(pca_2 = pca)

# merge datasets 

d4 <- bind_cols(d3, d3b)

# table for SNP results 

#Cluster 1 vs Cluster 1

d5 <- d4 %>% filter(pca_1 == "Cluster 1" & pca_2 == "Cluster 1")
d5b <- d5 %>% select(pca_1, dist) %>% rename(pca = pca_1)

a1 <- d5b %>% 
           summarize(mu = round(mean(dist)), sdx = round(sd(dist)), median = round(median(dist)), rangex = round(range(dist)))
a1
####################################
#Cluster 2 vs Cluster 2

d6 <- d4 %>% filter(pca_1 == "Cluster 2" & pca_2 == "Cluster 2")
d6b <- d6 %>% select(pca_1, dist) %>% rename(pca = pca_1)

a2 <- d6b %>% 
           summarize(mu = round(mean(dist)), sdx = round(sd(dist)), median = round(median(dist)), rangex = round(range(dist)))
a2
###########################################
#Cluster 1 vs Cluster 2

d7 <- d4 %>% filter(pca_1 == "Cluster 1" & pca_2 == "Cluster 2")
d7b <- d7 %>% select(pca_1, dist)
d7c <- d7b %>% mutate(pca = "Cluster_1_2")
d7c <- d7c %>% select(-pca_1)

#Cluster 2 vs Cluster 1

d8 <- d4 %>% filter(pca_1 == "Cluster 2" & pca_2 == "Cluster 1")
d8b <- d8 %>% select(pca_1, dist)
d8c <- d8b %>% mutate(pca = "Cluster_1_2")
d8c <- d8c %>% select(-pca_1)

d8d <- bind_rows(d7c, d8c)
a3 <- d8d %>% 
           summarize(mu = round(mean(dist)), sdx = round(sd(dist)), median = round(median(dist)), rangex = round(range(dist)))
a3
#########################################
# rowbind

d9 <- bind_rows(d5b, d6b, d7c, d8c)

# Use semi-transparent fill

snpinf <- ggplot(d9, aes(x = dist, fill = pca)) +
  geom_histogram() +
  theme_bw() + 
  theme(legend.position = "none") +
  xlab("Number of SNPs") +
  ylab("Frequency") +
  theme(axis.text.y = element_text(size = 30)) +
  theme(axis.title.y = element_text(size = 35, face = "bold")) +
  theme(axis.title.x = element_text(size = 35, face = "bold")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 30)) +
  theme(strip.text.x = element_text(size = 35)) +
  scale_fill_manual(values=c("darkblue", "darkgreen", "coral")) +
  facet_wrap(~pca)
snpinf
```