---
title: "Figure 6 newport"
author: "Joao Carlos Gomes-Neto"
date: "2/26/2021"
output: html_document
---

Load packages

```{r, include = FALSE}
# Load packages

library(broom)
library(tidyverse)
library(skimr)
library(RColorBrewer)
library(gridExtra)
library(lattice)
library(ggpubr)
library(vegan)
library(reshape2)
library(ggrepel)
library(ggnewscale)
library(forcats)
library(naniar)
library(data.table)
library(rARPACK)
library(logisticPCA)
library(cluster)
library(factoextra)
library(NbClust)
library(plotly)
library(ggridges)
```

Generating metadata for Salmonella Newport heatmap plot

```{r, include = FALSE}

# BAPS1 
# Enter Baps 1-6 

baps <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/newport/input_data/fastbaps_partition_baps_prior_l6.csv')

# Changing column names 

colnames(baps)[1:2] <- c("id", "baps_1") 

# Select colunms id and baps_1
b1 <- baps %>% 
              select(id, baps_1)


d1 <- b1

########################################################################
# ST

mlst <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/newport/input_data/salmonellast_output.csv')

# QC the data

skim(mlst)

# Generate the id column 

mlst$id2 <- sapply(strsplit(as.character(mlst$FILE),'_'), "[", 1)

# Select columns of interest

s1 <- mlst %>%
          select(id2, ST) %>%
            mutate_all(na_if, "-") %>%
              mutate_all(na_if, "?") %>%
                  rename(id = id2)

# st

s1$st <- ifelse(s1$ST == 5, "ST5", 
                   ifelse(s1$ST == 45, "ST45", 
                          ifelse(s1$ST == 118, "ST118", "Other STs")))

d2 <- s1

d2 <- d2 %>% select(-ST)
#########################################################################
# SISTR

# Enter SISTR results

sistr <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/newport/input_data/sistr_output.csv')

# QC the data 

skim(sistr)

# Generate the id column 

sistr$id <- sapply(strsplit(as.character(sistr$genome),'_'), "[", 1)

# Select columns 

sis1 <- sistr %>%
            select(id, cgmlst_ST, serovar_cgmlst) %>%
            mutate_all(na_if, "-") %>%
              mutate_all(na_if, "?") 

sis1$serovar <- ifelse(sis1$serovar_cgmlst == "Newport", "Newport",
                            "Other serovars")

sis1 <- mutate(sis1, cgmlst = ifelse(cgmlst_ST %in% 1468400426, "cgMLST 1468400426",
                                     ifelse(cgmlst_ST %in% 88443731, "cgMLST 88443731", 
                                            ifelse(cgmlst_ST %in% 1271156802, "cgMLST 1271156802",
                                            "Other cgMLSTs"))))
d3 <- sis1

############################################################

# KMER 
# enter the data

data1 <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/newport/kmer_matrix/kmer_newport.csv')

#  Non-supervised kmeans clustering analysis

cluster_data <- data1 %>% select(tSNE1, tSNE2)

# combining both approaches we determine that 2 clusters are enough 
set.seed(123)
km.final <- kmeans(cluster_data, 4)
data1$kmer_cluster <- km.final$cluster
data1$kmer_cluster <- as.factor(data1$kmer_cluster)

# select the column we need

d4 <- data1 %>% select(id, kmer_cluster)
###################################################################

# SNP 
# enter the data

data1 <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/newport/snp_matrix/snp_newport.csv')

#  Non-supervised kmeans clustering analysis

cluster_data <- data1 %>% select(tSNE1, tSNE2)

# combining both approaches we determine that 2 clusters are enough 
set.seed(123)
km.final <- kmeans(cluster_data, 3)
data1$snp_cluster <- km.final$cluster
data1$snp_cluster <- as.factor(data1$snp_cluster)

# filter the column

d5 <- data1 %>% select(id, snp_cluster)

#############################################################################################

# PCA CLUSTER

data1 <- read_csv("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/newport/pca_newport/pca_data2.csv")
data2 <- read_csv("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/newport/pca_newport/cluster2.csv")

d6 <- data1 %>% select(-X1)
d7 <- data2 %>% select(-X1)
#############################################################################################
#############################################################################################

# combining all datasets

d8 <- left_join(d1, d2, on = "id")
d9 <- left_join(d8, d3, on = "id")
d10 <- left_join(d9, d4, on = "id")
d11 <- left_join(d10, d5, on = "id")
d12 <- left_join(d11, d6, on = "id")
d13 <- left_join(d12, d7, on = "id")

# if NA for ST call others 

d14 <- d13 %>% mutate(st = replace_na(st, "Other STs"))
d15 <- d14 %>% mutate(pca_cluster = cluster) %>% select(-cluster)
d15 <- d15 %>% mutate(baps_1 = as.factor(baps_1))
d15 <- d15 %>% mutate(pca_cluster = as.factor(pca_cluster))
d15 <- d15 %>% select(id, baps_1, st, cgmlst, kmer_cluster, snp_cluster, pca_cluster)

#############################################################################################
#############################################################################################

# Getting the AMR data 

abx1 <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/newport/input_data/sabricate_resfinder_output.csv')
abx1$id <- sapply(strsplit(as.character(abx1$'#FILE'),'_'), "[", 1)
abx1 <- abx1 %>% mutate(gene = GENE)
abx2 <- abx1 %>% select(id, gene)

# eliminate genes with double or more counts

abx3 <- abx2 %>% group_by(id, gene) %>% count()
abx4 <- abx3 %>% filter(n <= 1)

# spread the data

abx5 <- spread(abx4, key = gene, value = n)

# input zeros for replacing NAs

abx5[is.na(abx5)] <- 0

# merge with d15

d16 <- left_join(d15, abx5, on = "id")
d17 <- column_to_rownames(d16, var = "id")

# export file as csv

newport_abx <- d17
write.csv(newport_abx, "~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/newport/input_data/newport_abx.csv")

# filter genes that we need

d18 <- newport_abx %>% select(baps_1, st, cgmlst, kmer_cluster, snp_cluster, pca_cluster,
                                             `aac(6')-Iaa_1`, `aph(3'')-Ib_5`, `aph(6)-Id_1`,
                                              `blaCMY-2_1`, floR_2, `mdf(A)_1`, sul2_2, `tet(A)_6`,
                                               aadA2_1, `ant(3'')-Ia_1`, `blaCARB-2_1`, dfrA1_8, `mph(A)_2`,
                                              qnrA1_1)

newport_abx_final <- d18
write.csv(newport_abx_final, "~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/newport/input_data/newport_abx_final.csv")

d18c <- rownames_to_column(d18, var = "id")
#############################################################################################
#############################################################################################
# Getting the plasmid data 

x1 <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/newport/input_data/sabricate_plasmidfinder_output.csv')
x1$id <- sapply(strsplit(as.character(x1$'#FILE'),'_'), "[", 1)
x1 <- x1 %>% mutate(gene = GENE)
x2 <- x1 %>% select(id, gene)

# eliminate genes with double or more counts

x3 <- x2 %>% group_by(id, gene) %>% count()
x4 <- x3 %>% filter(n <= 1)

# spread the data

x5 <- spread(x4, key = gene, value = n)

# input zeros for replacing NAs

x5[is.na(x5)] <- 0

# merge with d15

d16 <- left_join(d15, x5, on = "id")
d17 <- column_to_rownames(d16, var = "id")
d17[is.na(d17)] <- 0

# export file as csv

newport_plasmid <- d17
write.csv(newport_plasmid, "~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/newport/input_data/newport_plasmid.csv")

# filter plasmids that we need

d18b <- newport_plasmid %>% select(
                                   ColRNAI_1, `IncA/C2_1`, `IncFIC(FII)_1`, `IncFII(p96A)_1_p96A`,
                                   IncN_1, IncR_1, pENTAS02_1)

newport_plasmid_final <- d18b
write.csv(newport_plasmid_final, "~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/newport/input_data/newport_plasmid_final.csv")

d19b <- rownames_to_column(d18b, var = "id")

######################################################################################
######################################################################################
######################################################################################

# combine AMR and plasmids into one dataset

d20 <- left_join(d18c, d19b, on = "id")
d20 <- column_to_rownames(d20, var = "id")
write.csv(d20, "~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/newport/input_data/d20.csv")

#####################################################################################
#####################################################################################
# PANGEA results
# Enter empirical results

data1 <- read_csv("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/newport/gwas_genes/newport_st45_gene_gwas_filtered.csv")

# Find top predictors with 70% coverage and above

data2 <- data1%>% 
  filter(!grepl('group_', gene_kmer_id))

# get the list of genes

list_genes <- data2$gene_kmer_id

# Subset the original data based on that
data3 <- data1 %>%
      filter(gene_kmer_id %in% list_genes) 

# Create a column called empirical 

data3$results <- "Empirical"

####################################################

# Enter random results 

data4 <- read_csv("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/newport/gwas_genes/random_newport_st45_gene_gwas_not_filtered.csv")

# Subset the original data based on that
data5 <- data4 %>%
      filter(gene_kmer_id %in% list_genes) 

# Create a column called empirical 

data5$results <- "Random"

##################################################

# Merge empirical and random results

data6 <- rbind(data3, data5)
data6 <- rename(data6, Accuracy = accur)
data6$results <- factor(data6$results, levels=c("Random", "Empirical"))

# Plot data

p9b <- ggplot(data6, aes(x = pos_pred_value, y = gene_kmer_id, size = neg_pred_value, color = Accuracy)) +
  theme_bw() +
  xlab("Positive predictive value") +
  ylab("") +
  xlim(0, 1) +
  theme(axis.text.y = element_text(size = 30)) +
  theme(axis.title.y = element_text(size = 40, face = "bold")) +
  theme(axis.title.x = element_text(size = 40, face = "bold")) +
  theme(axis.text.x = element_text(size = 30, angle = 45, vjust = 1, hjust=1)) +
  theme(strip.text.x = element_text(size = 35, colour = "black", angle = 0)) +
  geom_point() + 
  ggtitle("ST45 vs Other STs (loci)") +  
  theme(plot.title = element_text(size = 50, face = "bold")) + 
  theme(legend.title=element_text(size=40, face = "bold")) +
  theme(legend.text=element_text(size=30)) +
  scale_size_continuous(name = "Negative predictive value") +
  scale_colour_gradient(low = "orange", high = "darkblue") +
  facet_wrap(~results)
p9b
################################################################################
################################################################################
################################################################################
# enter Rtab file and select genes needed

d1 <- read_tsv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/newport/input_data/gene_presence_absence.Rtab')

# set Genes as rownames

d1 <- column_to_rownames(d1, var = "Gene") 

d1$Gene <- NULL

# Transpose it

d2 <- transpose(d1)

# Add column and rownames

rownames(d2) <- colnames(d1)
colnames(d2) <- rownames(d1)

# Reset id to colnames

d3 <- rownames_to_column(d2, "id")

# List of targeted genes 

genes <- list_genes

######################################################################
genes <- c("symE_1", "ydhC_2", "neo", "hdfR_4", "folP_2", "soj_1", "addA", "noc", "higB2",
           "hupB_2", "topB_2", "xerC_2", "polC", "rhsC", "tus_2", "blc_2", "qacA", "qacB", "sugE", "sugE_1", "sugE_2",
           "caf1M_2", "caf1A_3")
d4 <- d3 %>% select("id", contains(genes))

remove_genes <- c("neo_2", "neo_3", "addA_2", "noc_1", "noc_2", "higB2_2", "polC_2")

d5 <- d4 %>% select(!remove_genes)

# merge with d20
d20 <- rownames_to_column(d20, var = "id")
d21 <- left_join(d20, d5, on = "id")
lociPopStr_st45 <- column_to_rownames(d21, var = "id")
write.csv(lociPopStr_st45, "~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/newport/input_data/lociPopStr_st45.csv")
```

Analysis of samples containing clinical bovine and human isolates

```{r, include = FALSE}

# Enter group1-Baps

# Enter the data, create a new column for group, and generate the id column
data4 <- read_csv("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/newport/unmc_genomes/fastbaps_baps.csv")
colnames(data4)[1:2] <- c("id", "baps_1")

d1 <- data4 %>% select(id, baps_1)
###########################################################################
# Enter MLST results

mlst <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/newport/unmc_genomes/mlst_output.csv')

# Generate the id column 

mlst$id <- sapply(strsplit(as.character(mlst$FILE),'_'), "[", 1)

# Create data1 

d2 <- mlst %>% select(id, ST)
d2 <- d2 %>% mutate(st = case_when(ST == 5 ~ "ST5",
                                           ST == 31 ~ "ST31",
                                           ST == 45 ~ "ST45",
                                           ST == 118 ~ "ST118"))

d2 <- d2 %>% select(id, st)
####################################################################

# Enter SISTR results

sistr <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/newport/unmc_genomes/sistr_all_merge.csv')

d3 <- sistr %>% select(genome, cgmlst_ST) %>% rename(id2 = genome)
d3$id <- sapply(strsplit(as.character(d3$id2),'_'), "[", 1)
d3 <- d3 %>% select(id, cgmlst_ST)
d3 <- d3 %>% mutate(cgmlst = case_when(cgmlst_ST == 88443731 ~ "cgMLST 88443731",
                                           cgmlst_ST == 1044935463 ~ "cgMLST 1044935463",
                                           cgmlst_ST ==  1416474556 ~ "cgMLST 1416474556",
                                           cgmlst_ST ==  1468400426 ~ "cgMLST 1468400426",
                                           cgmlst_ST ==  1677424419 ~ "cgMLST 1677424419",
                                           cgmlst_ST ==  2245200879 ~ "cgMLST 2245200879",
                                           cgmlst_ST ==  2996810515 ~ "cgMLST 2996810515",
                                           cgmlst_ST ==  3085959049 ~ "cgMLST 3085959049",
                                           cgmlst_ST ==  3336043520 ~ "cgMLST 3336043520",
                                           cgmlst_ST ==  3429497560 ~ "cgMLST 3429497560",
                                           cgmlst_ST ==  3659605560 ~ "cgMLST 3659605560",
                                           cgmlst_ST ==  3752976818 ~ "cgMLST 3752976818",
                                           cgmlst_ST ==  4038130518 ~ "cgMLST 4038130518"))
d3 <- d3 %>% select(id, cgmlst)

# merge datasets

d4 <- left_join(d1, d2, on = "id")
d5 <- left_join(d4, d3, on = "id")
d6 <- d5 %>% mutate(baps_1 = as.factor(baps_1))
write.csv(d5, file = "newport_unmc_isolates_cgmlst.csv", na = " ")

########################################################################
########################################################################

# Enter Rtab file and select genes of interest

# enter Rtab file and select genes needed

d1b <- read_tsv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/newport/unmc_genomes/gene_presence_absence.Rtab')

# set Genes as rownames

d1b <- column_to_rownames(d1b, var = "Gene") 

d1b$Gene <- NULL

# Transpose it

d2b <- transpose(d1b)

# Add column and rownames

rownames(d2b) <- colnames(d1b)
colnames(d2b) <- rownames(d1b)

# Reset id to colnames

d3b <- rownames_to_column(d2b, "id")

######################################################################
genes <- c("symE_1", "ydhC_2", "neo", "hdfR_4", "folP_2", "soj_1", "addA", "noc", "higB2",
           "hupB_2", "topB_2", "xerC_2", "polC", "rhsC", "tus_2", "blc_2", "qacA", "qacB", "sugE", "sugE_1", "sugE_2",
           "caf1M_2", "caf1A_3")
d4b <- d3b %>% select("id", contains(genes))

#remove_genes <- c("neo_2", "neo_3", "addA_2", "noc_1", "noc_2", "higB2_2", "polC_2")
#d5b <- d4b %>% select(!remove_genes)

# merge with d20

d8 <- left_join(d6, d4b, on = "id")
lociPopStr_st45_sample <- column_to_rownames(d8, var = "id")
write.csv(lociPopStr_st45_sample, "~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/newport/unmc_genomes/lociPopStr_st45_sample.csv")

#####################################################################################
#####################################################################################
#####################################################################################

data1 <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/newport/unmc_genomes/phenotype_newport.csv')

# Create a new column-based on presence of sugE_2

data1$ST <- ifelse(data1$id == 190211, "ST45 (sugE_2 pos)", 
                   ifelse(data1$id == 190502, "ST45 (sugE_2 pos)", 
                         ifelse(data1$id == 180410, "ST45 (sugE_2 neg)", 
                          ifelse(data1$id == 190358, "ST45 (sugE_2 neg)", 
                          ifelse(data1$id == 180603, "ST118 (sugE_2 neg)", 
                          ifelse(data1$id == 190454, "ST118 (sugE_2 neg)", 
                          ifelse(data1$id == 180244, "ST118 (sugE_2 neg)",      
                          ifelse(data1$id == 180486, "ST45 (sugE_2 neg)", 
                          ifelse(data1$id == 190013, "ST118 (sugE_2 neg)",
                          ifelse(data1$id == 180537, "ST5 (sugE_2 neg)", 
                          ifelse(data1$id == 190470, "ST5 (sugE_2 neg)", "ST31 (sugE_2 neg)")))))))))))

data1$treatment <- ifelse(data1$concentration == "0", "Cetylpyridinium chloride (0 ug/mL)", 
                          ifelse(data1$concentration == "15", "Cetylpyridinium chloride (15 ug/mL)", 
                          ifelse(data1$concentration == "20", "Cetylpyridinium chloride (20 ug/mL)", 
                         "Cetylpyridinium chloride (25 ug/mL)")))
# Filter out ids that we don't need

#data2 <- data1 %>% filter(id != 180244 & id != 190013 & id != 180764)


# plot the data
data2 <- data1 %>% filter(treatment == "Cetylpyridinium chloride (0 ug/mL)" | treatment == "Cetylpyridinium chloride (25 ug/mL)")
data2 <- data2 %>% mutate(Treatments = treatment)
data2$ST <- factor(data2$ST, levels=c("ST5 (sugE_2 neg)", "ST118 (sugE_2 neg)", "ST31 (sugE_2 neg)",
                                      "ST45 (sugE_2 neg)", "ST45 (sugE_2 pos)"))

fig10c <- ggplot(data2, aes(x = time, y = absorbance, color = Treatments)) +
    theme_bw() +
     theme(legend.position = "none") +
    xlab("Time (hours)") +
    ylab("OD values") +
    theme(axis.text.y = element_text(size = 20)) +
    theme(axis.title.y = element_text(size = 30, face = "bold")) +
    theme(axis.title.x = element_text(size = 30, face = "bold")) +
    theme(axis.text.x = element_text(size = 20, angle = 0, vjust = 1, hjust=0.5)) +
    theme(strip.text.x = element_text(size = 20, colour = "black", angle = 0, face = "bold")) +
    theme(legend.title=element_text(size=40, face = "bold")) +
    theme(legend.text=element_text(size=25)) +
   # geom_boxplot(aes(color = ST)) +
    ggtitle("") +  
    geom_hline(yintercept = 0.6, linetype="dashed", color = "black", size = 1.5) +
    #labs(color = "ST") +
    facet_wrap(~ST, ncol = 5) +
    scale_color_manual(values = c("darkgreen", "steelblue")) +
    scale_fill_manual(values = c("darkgreen", "steelblue")) +
     geom_smooth(aes(group = Treatments, color = Treatments, fill = Treatments), method = "loess", se = TRUE, size = 0.5)
fig10c

# plot again to get legend

# plot the data
data2 <- data1 %>% filter(treatment == "Cetylpyridinium chloride (0 ug/mL)" | treatment == "Cetylpyridinium chloride (25 ug/mL)")
data2 <- data2 %>% mutate(Treatments = treatment)
data2$ST <- factor(data2$ST, levels=c("ST5 (sugE_2 neg)", "ST118 (sugE_2 neg)", "ST31 (sugE_2 neg)",
                                      "ST45 (sugE_2 neg)", "ST45 (sugE_2 pos)"))

fig10d <- ggplot(data2, aes(x = time, y = absorbance, color = Treatments)) +
    theme_bw() +
     #theme(legend.position = "none") +
    xlab("Time (hours)") +
    ylab("OD values") +
    theme(axis.text.y = element_text(size = 20)) +
    theme(axis.title.y = element_text(size = 30, face = "bold")) +
    theme(axis.title.x = element_text(size = 30, face = "bold")) +
    theme(axis.text.x = element_text(size = 20, angle = 0, vjust = 1, hjust=0.5)) +
    theme(strip.text.x = element_text(size = 20, colour = "black", angle = 0, face = "bold")) +
    theme(legend.title=element_text(size=40, face = "bold")) +
    theme(legend.text=element_text(size=25)) +
   # geom_boxplot(aes(color = ST)) +
    ggtitle("") +  
    geom_hline(yintercept = 0.6, linetype="dashed", color = "black", size = 1.5) +
    #labs(color = "ST") +
    facet_wrap(~ST, ncol = 5) +
    scale_color_manual(values = c("darkgreen", "steelblue")) +
    scale_fill_manual(values = c("darkgreen", "steelblue")) +
     geom_smooth(aes(group = Treatments, color = Treatments, fill = Treatments), method = "loess", se = TRUE, size = 0.5)
fig10d

# multiple regression model

model1 <- aov(absorbance ~ ST*Treatments*time, data = data2)
summary(model1)

#####################################################################

# slope based analysis (time from 6 to 18 hours) - rate of growth in the log phase

data3 <- data2 %>% filter(Treatments == "Cetylpyridinium chloride (25 ug/mL)")
data4 <- data3 %>% filter(time > 4 & time < 20)

fig10d <- ggplot(data4, aes(x = time, y = absorbance, color = ST)) +
    theme_bw() +
     #theme(legend.position = "none") +
    xlab("Time (hours)") +
    ylab("OD values") +
    theme(axis.text.y = element_text(size = 20)) +
    theme(axis.title.y = element_text(size = 30, face = "bold")) +
    theme(axis.title.x = element_text(size = 30, face = "bold")) +
    theme(axis.text.x = element_text(size = 20, angle = 0, vjust = 1, hjust=0.5)) +
    theme(strip.text.x = element_text(size = 20, colour = "black", angle = 0, face = "bold")) +
    theme(legend.title=element_text(size=40, face = "bold")) +
    theme(legend.text=element_text(size=25)) +
   # geom_boxplot(aes(color = ST)) +
    ggtitle("") +  
    #labs(color = "ST") +
    scale_color_manual(values = c("darkgreen", "darkblue", "orange", "purple", "gray")) +
    scale_fill_manual(values = c("darkgreen", "darkblue", "orange", "purple", "gray")) +
     geom_smooth(aes(group = ST, color = ST, fill = ST), method = "lm", se = TRUE, size = 0.5)
fig10d

# get one linear regression per group

data5 <- data4 %>% select(absorbance, time, ST)
model2 <- data5 %>% group_by(ST) %>%
  do(fitreg = tidy(glm(absorbance ~ time, data = .))) %>%
  unnest(fitreg)
model2

# lets get the final population size
data3 <- data2 %>% filter(Treatments == "Cetylpyridinium chloride (25 ug/mL)")
data4 <- data3 %>% filter(time == 0 | time == 24)
data5 <- data4 %>% select(id, time, absorbance, ST)
data7 <- data5 %>% filter(time == 0)
data8 <- data5 %>% filter(time == 24) %>% select(-ST)
data9 <- bind_cols(data7, data8)
data10 <- data9 %>% select(-"id...5")
data10 <- data10 %>% rename(id = "id...1", time0 = "time...2",
                            od1 = "absorbance...3", ST = "ST", time24 = "time...6", od24 = "absorbance...7")
data11 <- data10 %>% mutate(prop = od24/od1)

# plot data

p1 <- ggplot(data5, aes(x = absorbance, y = ST, fill = ST)) +
    theme_bw() +
    xlim(0, 1) +
    theme(legend.position = "none") +
    xlab("OD values") +
    ylab("") +
    theme(axis.title.x = element_text(size = 35, face = "bold")) +
    theme(axis.text.x = element_text(size = 30, angle = 0, vjust = 1, hjust=0.5)) +
      theme(axis.text.y = element_text(size = 30)) +
    scale_fill_manual(values = c("purple", "orange", "gray", "steelblue", "darkblue")) +
  ggtitle("Growth using Cetylpyridinium chloride (25 ug/mL)") + 
theme(plot.title = element_text(size=30, face = "bold")) +   geom_boxplot()
p1

# plot data

p1 <- ggplot(data11, aes(x = prop, y = ST, fill = ST)) +
    theme_bw() +
    xlim(0, 20) +
    theme(legend.position = "none") +
    xlab("OD values") +
    ylab("") +
    theme(axis.title.x = element_text(size = 35, face = "bold")) +
    theme(axis.text.x = element_text(size = 30, angle = 0, vjust = 1, hjust=0.5)) +
      theme(axis.text.y = element_text(size = 30)) +
    scale_fill_manual(values = c("purple", "orange", "gray", "steelblue", "darkblue")) +
  ggtitle("Growth using Cetylpyridinium chloride (25 ug/mL)") + 
theme(plot.title = element_text(size=30, face = "bold")) +   geom_violin()
p1

a <- ggplot(data11, aes(x = prop, y = ST)) +
  geom_density_ridges(aes(fill = ST)) +
  scale_fill_manual(values = c("purple", "orange", "gray", "steelblue", "darkblue")) +
  theme(legend.position = "none")
a

# plot distribution and quartiles
a <- ggplot(data11, aes(x = prop, y = ST, fill = factor(stat(quantile)))) +
  stat_density_ridges(
    geom = "density_ridges_gradient", calc_ecdf = TRUE,
    quantiles = 4, quantile_lines = TRUE
  ) +
  theme_minimal() +
  scale_fill_viridis_d(name = "Quartiles") +
  theme(axis.title.x = element_text(size = 35, face = "bold")) +
    theme(axis.text.x = element_text(size = 30, angle = 0, vjust = 1, hjust=0.5)) +
      theme(axis.text.y = element_text(size = 30)) +
  xlab("Growth ratio (OD 24/OD 0)") +
  theme(legend.title=element_text(size=40, face = "bold")) +
    theme(legend.text=element_text(size=30)) +
  ggtitle("Growth using Cetylpyridinium chloride (25 ug/mL)") + 
theme(plot.title = element_text(size=30, face = "bold")) +
  ylab("") +
  xlim(0, 25)
a


# run an anova model on it

mod1 <- aov(prop ~ ST, data = data11)
summary(mod1)

# pairwise test

test <- pairwise.t.test(data11$prop, data11$ST, p.adj = "none")
test

```