---
title: "figure_4_sal_paper"
author: "Joao Carlos Gomes-Neto"
date: "1/8/2021"
output: html_document
---

Load packages

```{r, include = FALSE}
# Load packages

library(tidyverse)
library(skimr)
library(RColorBrewer)
library(gridExtra)
library(lattice)
library(ggpubr)
library(vegan)
library(reshape2)
library(ggrepel)
library(ggnewscale)
library(forcats)
library(naniar)
library(data.table)
library(rARPACK)
library(logisticPCA)
library(cluster)
library(factoextra)
library(NbClust)
library(plotly)
```

Salmonella Infantis plots 

```{r, include = FALSE}
# BAPS1

baps <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/input_data/fastbaps_partition_baps_prior_l6.csv')

# Changing column names 

colnames(baps)[1:2] <- c("id", "baps_1") 

# Select colunms id and baps_1
b1 <- baps %>% 
              select(id, baps_1)


d1 <- b1

#############################################################################################

# ST

# enter mlst data 
mlst <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/input_data/salmonellast_output.csv')

# QC the data

skim(mlst)

# Generate the id column 

mlst$id2 <- sapply(strsplit(as.character(mlst$FILE),'_'), "[", 1)

# Select columns of interest

s1 <- mlst %>%
          select(id2, ST) %>%
            mutate_all(na_if, "-") %>%
              mutate_all(na_if, "?") %>%
                  rename(id = id2)

# st

s1$st <- ifelse(s1$ST == 32, "ST32", "Other STs")

d2 <- s1

d2 <- d2 %>% select(-ST)

#############################################################################################

# CGMLST

# Enter SISTR results

sistr <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/input_data/sistr_all.csv')

# QC the data 

skim(sistr)

# Generate the id column 

sistr$id <- sapply(strsplit(as.character(sistr$genome),'_'), "[", 1)

# Select columns 

sis1 <- sistr %>%
            select(id, cgmlst_ST, serovar_cgmlst) %>%
            mutate_all(na_if, "-") %>%
              mutate_all(na_if, "?") 

sis1$serovar <- ifelse(sis1$serovar_cgmlst == "Infantis", "Infantis",
                            "Other serovars")

sis1 <- mutate(sis1, cgmlst = ifelse(cgmlst_ST %in% 2242423463, "cgMLST 2242423463",
                                            "Other cgMLSTs"))

d3 <- sis1

d3 <- d3 %>% select(id, cgmlst)

#############################################################################################

# KMER CLUSTER 

# Enter kmer-matrix

data1 <- read_csv("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/kmer_matrix/kmer_infantis.csv")

# generate cluster membership

#  Non-supervised kmeans clustering analysis

cluster_data <- data1 %>% select(tSNE1, tSNE2)

# combining both approaches we determine that 2 clusters are enough 
set.seed(123)
km.final <- kmeans(cluster_data, 2)
data1$kmer_cluster <- km.final$cluster
data1$kmer_cluster <- as.factor(data1$kmer_cluster)

# select the column we need

d4 <- data1 %>% select(id, kmer_cluster)

#############################################################################################

# SNP CLUSTER

# enter snp matrix

data2 <- read_csv("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/snp_matrix/snp_infantis.csv")

#  Non-supervised kmeans clustering analysis

cluster_data <- data2 %>% select(tSNE1, tSNE2)

# combining both approaches we determine that 2 clusters are enough 
set.seed(123)
km.final <- kmeans(cluster_data, 3)
data2$snp_cluster <- km.final$cluster
data2$snp_cluster <- as.factor(data2$snp_cluster)

# filter the column

d5 <- data2 %>% select(id, snp_cluster)

#############################################################################################

# PCA CLUSTER

data1 <- read_csv("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/pca_infantis/pca_data2.csv")
data2 <- read_csv("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/pca_infantis/cluster2.csv")

d6 <- data1 %>% select(-X1)
d7 <- data2 %>% select(-X1)
#############################################################################################
#############################################################################################

# combining all datasets

d8 <- left_join(d1, d2, on = "id")
d9 <- left_join(d8, d3, on = "id")
d10 <- left_join(d9, d4, on = "id")
d11 <- left_join(d10, d5, on = "id")
d12 <- left_join(d11, d6, on = "id")
d13 <- left_join(d12, d7, on = "id")

# if NA for ST call others 

d14 <- d13 %>% mutate(st = replace_na(st, "Other STs"))
d15 <- d14 %>% mutate(pca_cluster = cluster) %>% select(-cluster)
d15 <- d15 %>% mutate(baps_1 = as.factor(baps_1))
d15 <- d15 %>% mutate(pca_cluster = as.factor(pca_cluster))

#############################################################################################

# Plot baps1 onto accessory genome

d15$baps_1 <- factor(d15$baps_1, levels=c("1", "2", "3"))
p1 <- ggplot(d15, aes(x = PC1, y = PC2, color = baps_1)) + 
  theme_bw() + 
  xlab("PC1") +
  ylab("PC2") +
  theme(axis.text.y = element_text(size = 40)) +
  theme(axis.title.y = element_text(size = 45, face = "bold")) +
  theme(axis.title.x = element_text(size = 45, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 1, size = 40)) +
  scale_color_manual(values=c("darkred", "purple", "darkgreen")) +
   ggtitle("A.") +
  theme(plot.title = element_text(size = 50, face = "bold")) + 
  labs(color='BAPS1') + 
  geom_jitter(pch = 19, height = 1, width = 1, size = 4) +
  theme(legend.title=element_text(size=50, face = "bold")) +
  theme(legend.text=element_text(size=40)) +
  guides(color = guide_legend(override.aes = list(size=20)))
p1

# Plot st onto accessory genome

d15$st <- factor(d15$st, levels=c("ST32","Other STs"))
p2 <- ggplot(d15, aes(x = PC1, y = PC2, color = st)) + 
  theme_bw() + 
  xlab("PC1") +
  ylab("PC2") +
  theme(axis.text.y = element_text(size = 40)) +
  theme(axis.title.y = element_text(size = 45, face = "bold")) +
  theme(axis.title.x = element_text(size = 45, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 1, size = 40)) +
  scale_color_manual(values=c("darkred", "gray")) +
   ggtitle("B.") +
  theme(plot.title = element_text(size = 50, face = "bold")) + 
  labs(color='ST') + 
  geom_jitter(pch = 19, height = 1, width = 1, size = 4) +
  theme(legend.title=element_text(size=50, face = "bold")) +
  theme(legend.text=element_text(size=40)) +
  guides(color = guide_legend(override.aes = list(size=20)))
p2

# Plot cgmlst onto accessory genome

d15$cgmlst <- factor(d15$cgmlst, levels=c("cgMLST 2242423463",
                                            "Other cgMLSTs"))
p3 <- ggplot(d15, aes(x = PC1, y = PC2, color = cgmlst)) + 
  theme_bw() + 
  xlab("PC1") +
  ylab("PC2") +
  theme(axis.text.y = element_text(size = 40)) +
  theme(axis.title.y = element_text(size = 45, face = "bold")) +
  theme(axis.title.x = element_text(size = 45, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 1, size = 40)) +
  scale_color_manual(values=c("black", "orange")) +
   ggtitle("C.") +
  theme(plot.title = element_text(size = 50, face = "bold")) + 
  labs(color='cgMLST') + 
  geom_jitter(pch = 19, height = 1, width = 1, size = 4) +
  theme(legend.title=element_text(size=50, face = "bold")) +
  theme(legend.text=element_text(size=40)) +
  guides(color = guide_legend(override.aes = list(size=20)))
p3

# Plot kmer_cluster onto accessory genome

p4 <- ggplot(d15, aes(x = PC1, y = PC2, color = kmer_cluster)) + 
  theme_bw() + 
  xlab("PC1") +
  ylab("PC2") +
  theme(axis.text.y = element_text(size = 40)) +
  theme(axis.title.y = element_text(size = 45, face = "bold")) +
  theme(axis.title.x = element_text(size = 45, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 1, size = 40)) +
  scale_color_manual(values=c("gray", "darkred")) +
   ggtitle("D.") +
  theme(plot.title = element_text(size = 50, face = "bold")) + 
  labs(color='Kmer-clusters') + 
  geom_jitter(pch = 19, height = 1, width = 1, size = 4) +
  theme(legend.title=element_text(size=50, face = "bold")) +
  theme(legend.text=element_text(size=40)) +
  guides(color = guide_legend(override.aes = list(size=20)))
p4

# Plot snp_cluster onto accessory genome 

p5 <- ggplot(d15, aes(x = PC1, y = PC2, color = snp_cluster)) + 
  theme_bw() + 
  xlab("PC1") +
  ylab("PC2") +
  theme(axis.text.y = element_text(size = 40)) +
  theme(axis.title.y = element_text(size = 45, face = "bold")) +
  theme(axis.title.x = element_text(size = 45, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 1, size = 40)) +
  scale_color_manual(values=c("gray", "darkred", "darkblue")) +
   ggtitle("E.") +
  theme(plot.title = element_text(size = 50, face = "bold")) + 
  labs(color='SNP-clusters') + 
  geom_jitter(pch = 19, height = 1, width = 1, size = 4) +
  theme(legend.title=element_text(size=50, face = "bold")) +
  theme(legend.text=element_text(size=40)) +
  guides(color = guide_legend(override.aes = list(size=20)))
p5

# Plot pca_cluster onto accessory genome 

p6 <- ggplot(d15, aes(x = PC1, y = PC2, color = pca_cluster)) + 
  theme_bw() + 
  xlab("PC1") +
  ylab("PC2") +
  theme(axis.text.y = element_text(size = 40)) +
  theme(axis.title.y = element_text(size = 45, face = "bold")) +
  theme(axis.title.x = element_text(size = 45, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 1, size = 40)) +
  scale_color_manual(values=c("darkblue", "darkred")) +
   ggtitle("F.") +
  theme(plot.title = element_text(size = 50, face = "bold")) + 
  labs(color='PCA-clusters') + 
  geom_jitter(pch = 19, height = 1, width = 1, size = 4) +
  theme(legend.title=element_text(size=50, face = "bold")) +
  theme(legend.text=element_text(size=40)) +
  guides(color = guide_legend(override.aes = list(size=20)))
p6

# examine how many clusters are there for infantis

a <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/pca_infantis/wss2.csv')

p7 <- ggplot(a, aes(x = X1, y = wss)) + geom_line() + xlab("Clusters") + ylab("Within cluster sum of squares") +
  scale_x_continuous(breaks = c(seq(from = 0, to = 10, by = 1))) +
  theme(axis.text.y = element_text(size = 40)) +
  theme(axis.title.y = element_text(size = 45, face = "bold")) +
  theme(axis.title.x = element_text(size = 45, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 1, size = 40)) 
p7

# get the variance explained

b <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/pca_infantis/deviance_model.csv')
b

######################################################

# create file for Infantis pangea analysis

data2 <- read_csv("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/pca_infantis/cluster2.csv")
data3 <- data2 %>% select(id, cluster)

# create the phenotype column

data4 <- data3 %>% mutate(phenotype = ifelse(cluster == 1, 1, 0))

# select columnns needed 

data5 <- data4 %>% select(-cluster)

# save data as csv

phenotype_pangea_infantis <- data5
write.csv(phenotype_pangea_infantis, "~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/input_data/phenotype_pangea_infantis.csv")
```

Salmonella Newport plots

```{r, include = FALSE}

# BAPS1 
# Enter Baps 1-6 

baps <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/newport/input_data/fastbaps_partition_baps_prior_l6.csv')

# Changing column names 

colnames(baps)[1:2] <- c("id", "baps_1") 

# Select colunms id and baps_1
b1 <- baps %>% 
              select(id, baps_1)


d1 <- b1

########################################################################
# ST

mlst <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/newport/input_data/salmonellast_output.csv')

# QC the data

skim(mlst)

# Generate the id column 

mlst$id2 <- sapply(strsplit(as.character(mlst$FILE),'_'), "[", 1)

# Select columns of interest

s1 <- mlst %>%
          select(id2, ST) %>%
            mutate_all(na_if, "-") %>%
              mutate_all(na_if, "?") %>%
                  rename(id = id2)

# st

s1$st <- ifelse(s1$ST == 5, "ST5", 
                   ifelse(s1$ST == 45, "ST45", 
                          ifelse(s1$ST == 118, "ST118", "Other STs")))

d2 <- s1

d2 <- d2 %>% select(-ST)
#########################################################################
# SISTR

# Enter SISTR results

sistr <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/newport/input_data/sistr_output.csv')

# QC the data 

skim(sistr)

# Generate the id column 

sistr$id <- sapply(strsplit(as.character(sistr$genome),'_'), "[", 1)

# Select columns 

sis1 <- sistr %>%
            select(id, cgmlst_ST, serovar_cgmlst) %>%
            mutate_all(na_if, "-") %>%
              mutate_all(na_if, "?") 

sis1$serovar <- ifelse(sis1$serovar_cgmlst == "Newport", "Newport",
                            "Other serovars")

sis1 <- mutate(sis1, cgmlst = ifelse(cgmlst_ST %in% 1468400426, "cgMLST 1468400426",
                                     ifelse(cgmlst_ST %in% 88443731, "cgMLST 88443731", 
                                            ifelse(cgmlst_ST %in% 1271156802, "cgMLST 1271156802",
                                            "Other cgMLSTs"))))
d3 <- sis1

############################################################

# KMER 
# enter the data

data1 <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/newport/kmer_matrix/kmer_newport.csv')

#  Non-supervised kmeans clustering analysis

cluster_data <- data1 %>% select(tSNE1, tSNE2)

# combining both approaches we determine that 2 clusters are enough 
set.seed(123)
km.final <- kmeans(cluster_data, 4)
data1$kmer_cluster <- km.final$cluster
data1$kmer_cluster <- as.factor(data1$kmer_cluster)

# select the column we need

d4 <- data1 %>% select(id, kmer_cluster)
###################################################################

# SNP 
# enter the data

data1 <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/newport/snp_matrix/snp_newport.csv')

#  Non-supervised kmeans clustering analysis

cluster_data <- data1 %>% select(tSNE1, tSNE2)

# combining both approaches we determine that 2 clusters are enough 
set.seed(123)
km.final <- kmeans(cluster_data, 3)
data1$snp_cluster <- km.final$cluster
data1$snp_cluster <- as.factor(data1$snp_cluster)

# filter the column

d5 <- data1 %>% select(id, snp_cluster)

#############################################################################################

# PCA CLUSTER

data1 <- read_csv("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/newport/pca_newport/pca_data2.csv")
data2 <- read_csv("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/newport/pca_newport/cluster2.csv")

d6 <- data1 %>% select(-X1)
d7 <- data2 %>% select(-X1)
#############################################################################################
#############################################################################################

# combining all datasets

d8 <- left_join(d1, d2, on = "id")
d9 <- left_join(d8, d3, on = "id")
d10 <- left_join(d9, d4, on = "id")
d11 <- left_join(d10, d5, on = "id")
d12 <- left_join(d11, d6, on = "id")
d13 <- left_join(d12, d7, on = "id")

# if NA for ST call others 

d14 <- d13 %>% mutate(st = replace_na(st, "Other STs"))
d15 <- d14 %>% mutate(pca_cluster = cluster) %>% select(-cluster)
d15 <- d15 %>% mutate(baps_1 = as.factor(baps_1))
d15 <- d15 %>% mutate(pca_cluster = as.factor(pca_cluster))

#############################################################################################

# Plot baps1 onto accessory genome

d15$baps_1 <- factor(d15$baps_1, levels=c("1", "2", "3", "4", "5", "6", "7", "8", "9"))
p1 <- ggplot(d15, aes(x = PC1, y = PC2, color = baps_1)) + 
  theme_bw() + 
  xlab("PC1") +
  ylab("PC2") +
  theme(axis.text.y = element_text(size = 40)) +
  theme(axis.title.y = element_text(size = 45, face = "bold")) +
  theme(axis.title.x = element_text(size = 45, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 1, size = 40)) +
  scale_color_manual(values=c("cornflowerblue", "darkgreen", "gray", "darkred", "darkblue", "darkcyan",
                              "black", "orange", "brown")) +
   ggtitle("G.") +
  theme(plot.title = element_text(size = 50, face = "bold")) + 
  labs(color='BAPS1') + 
  geom_jitter(pch = 19, height = 1, width = 1, size = 4) +
  theme(legend.title=element_text(size=50, face = "bold")) +
  theme(legend.text=element_text(size=40)) +
  guides(color = guide_legend(override.aes = list(size=20)))
p1

# Plot st onto accessory genome

d15$st <- factor(d15$st, levels=c("ST5", "ST45", "ST118", "Other STs"))
p2 <- ggplot(d15, aes(x = PC1, y = PC2, color = st)) + 
  theme_bw() + 
  xlab("PC1") +
  ylab("PC2") +
  theme(axis.text.y = element_text(size = 40)) +
  theme(axis.title.y = element_text(size = 45, face = "bold")) +
  theme(axis.title.x = element_text(size = 45, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 1, size = 40)) +
  scale_color_manual(values=c("orange", "darkblue", "purple", "darkgreen")) +
   ggtitle("H.") +
  theme(plot.title = element_text(size = 50, face = "bold")) + 
  labs(color='ST') + 
  geom_jitter(pch = 19, height = 1, width = 1, size = 4) +
  theme(legend.title=element_text(size=50, face = "bold")) +
  theme(legend.text=element_text(size=40)) +
  guides(color = guide_legend(override.aes = list(size=20)))
p2

# Plot cgmlst onto accessory genome

d15$cgmlst <- factor(d15$cgmlst, levels=c("cgMLST 1468400426", "cgMLST 88443731", "cgMLST 1271156802", "Other cgMLSTs"))
p3 <- ggplot(d15, aes(x = PC1, y = PC2, color = cgmlst)) + 
  theme_bw() + 
  xlab("PC1") +
  ylab("PC2") +
  theme(axis.text.y = element_text(size = 40)) +
  theme(axis.title.y = element_text(size = 45, face = "bold")) +
  theme(axis.title.x = element_text(size = 45, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 1, size = 40)) +
  scale_color_manual(values=c("darkred", "darkblue", "darkgreen", "gray")) +
   ggtitle("I.") +
  theme(plot.title = element_text(size = 50, face = "bold")) + 
  labs(color='cgMLST') + 
  geom_jitter(pch = 19, height = 1, width = 1, size = 4) +
  theme(legend.title=element_text(size=50, face = "bold")) +
  theme(legend.text=element_text(size=40)) +
  guides(color = guide_legend(override.aes = list(size=20)))
p3

# Plot kmer_cluster onto accessory genome

p4 <- ggplot(d15, aes(x = PC1, y = PC2, color = kmer_cluster)) + 
  theme_bw() + 
  xlab("PC1") +
  ylab("PC2") +
  theme(axis.text.y = element_text(size = 40)) +
  theme(axis.title.y = element_text(size = 45, face = "bold")) +
  theme(axis.title.x = element_text(size = 45, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 1, size = 40)) +
  scale_color_manual(values=c("gray", "darkred", "steelblue", "purple")) +
   ggtitle("J.") +
  theme(plot.title = element_text(size = 50, face = "bold")) + 
  labs(color='Kmer-clusters') + 
  geom_jitter(pch = 19, height = 1, width = 1, size = 4) +
  theme(legend.title=element_text(size=50, face = "bold")) +
  theme(legend.text=element_text(size=40)) +
  guides(color = guide_legend(override.aes = list(size=20)))
p4

# Plot snp_cluster onto accessory genome 

p5 <- ggplot(d15, aes(x = PC1, y = PC2, color = snp_cluster)) + 
  theme_bw() + 
  xlab("PC1") +
  ylab("PC2") +
  theme(axis.text.y = element_text(size = 40)) +
  theme(axis.title.y = element_text(size = 45, face = "bold")) +
  theme(axis.title.x = element_text(size = 45, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 1, size = 40)) +
  scale_color_manual(values=c("gray", "darkred", "steelblue")) +
   ggtitle("K.") +
  theme(plot.title = element_text(size = 50, face = "bold")) + 
  labs(color='SNP-clusters') + 
  geom_jitter(pch = 19, height = 1, width = 1, size = 4) +
  theme(legend.title=element_text(size=50, face = "bold")) +
  theme(legend.text=element_text(size=40)) +
  guides(color = guide_legend(override.aes = list(size=20)))
p5

# Plot pca_cluster onto accessory genome 

p6 <- ggplot(d15, aes(x = PC1, y = PC2, color = pca_cluster)) + 
  theme_bw() + 
  xlab("PC1") +
  ylab("PC2") +
  theme(axis.text.y = element_text(size = 40)) +
  theme(axis.title.y = element_text(size = 45, face = "bold")) +
  theme(axis.title.x = element_text(size = 45, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 1, size = 40)) +
  scale_color_manual(values=c("gray", "darkred", "darkblue")) +
   ggtitle("L.") +
  theme(plot.title = element_text(size = 50, face = "bold")) + 
  labs(color='PCA-clusters') + 
  geom_jitter(pch = 19, height = 1, width = 1, size = 4) +
  theme(legend.title=element_text(size=50, face = "bold")) +
  theme(legend.text=element_text(size=40)) +
  guides(color = guide_legend(override.aes = list(size=20)))
p6

# examine how many clusters are there for newport

a <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/newport/pca_newport/wss2.csv')

p7 <- ggplot(a, aes(x = X1, y = wss)) + geom_line() + xlab("Clusters") + ylab("Within cluster sum of squares") +
  scale_x_continuous(breaks = c(seq(from = 0, to = 10, by = 1))) +
  theme(axis.text.y = element_text(size = 40)) +
  theme(axis.title.y = element_text(size = 45, face = "bold")) +
  theme(axis.title.x = element_text(size = 45, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 1, size = 40)) 
p7

# get the variance explained

b <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/newport/pca_newport/deviance_model.csv')
b
```

Salmonella Typhimurium plots 

```{r, include = FALSE}

# Enter group1-Baps

# Enter the data, create a new column for group, and generate the id column
data4 <- read_csv("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/group_1/input_data/roary_output_gr1.fastbaps_l6.csv")
data4$group <- "group_1"
colnames(data4)[1:2] <- c("id", "baps_1")

d1 <- data4 %>% select(id, baps_1, group)
###########################################################################
# Enter MLST results

mlst <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/global_input_data/salmonellast_output.csv')

# QC the data

skim(mlst)

# Generate the id column 

mlst$id2 <- sapply(strsplit(as.character(mlst$FILE),'/'), "[", 3)
s1 <- mlst %>% separate(id2, c("id", "day"), sep = "\\.")

# Select columns of interest

s2 <- s1 %>%
          select(id, ST) %>%
            mutate_all(na_if, "-") %>%
              mutate_all(na_if, "?")

# QC the data

skim(s2)

# st

s2$st <- ifelse(s2$ST == 19, "ST19", 
                   ifelse(s2$ST == 34, "ST34", 
                          ifelse(s2$ST == 36, "ST36",
                                 ifelse(s2$ST == 313, "ST313", "Other STs"))))

# Create data1 

d2 <- s2 

####################################################################

# Enter SISTR results

sistr <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/global_input_data/sistr_all.csv')

# QC the data 

skim(sistr)

# Select columns 

sis1 <- sistr %>%
            select(genome, cgmlst_ST, serovar_cgmlst) %>%
            mutate_all(na_if, "-") %>%
              mutate_all(na_if, "?") %>%
                rename(id = genome)

d3 <- sis1 

d3 <- mutate(d3, cgmlst = ifelse(cgmlst_ST %in% 1652656062, "cgMLST 1652656062",
                                     ifelse(cgmlst_ST %in% 860079270, "cgMLST 860079270", "Other cgMLSTs")))
##########################################################################
# KMER

data1 <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/group_1/kmer_matrix/kmer_group_1_typhimurium.csv')

#  Non-supervised kmeans clustering analysis

cluster_data <- data1 %>% select(tSNE1, tSNE2)

# combining both approaches we determine that 2 clusters are enough 
set.seed(123)
km.final <- kmeans(cluster_data, 6)
data1$kmer_cluster <- km.final$cluster
data1$kmer_cluster <- as.factor(data1$kmer_cluster)

# select the column we need

d4 <- data1 %>% select(id, kmer_cluster)
##################################################################################

# SNP

data1 <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/group_1/snp_matrix/snp_group_1_typhimurium.csv')

#  Non-supervised kmeans clustering analysis

cluster_data <- data1 %>% select(tSNE1, tSNE2)

# combining both approaches we determine that 2 clusters are enough 
set.seed(123)
km.final <- kmeans(cluster_data, 6)
data1$snp_cluster <- km.final$cluster
data1$snp_cluster <- as.factor(data1$snp_cluster)

# filter the column

d5 <- data1 %>% select(id, snp_cluster)

####################################################################################
#############################################################################################

# PCA CLUSTER

data1 <- read_csv("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/group_1/pca_group_1/pca_data2.csv")
data2 <- read_csv("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/group_1/pca_group_1/cluster2.csv")

d6 <- data1 %>% select(-X1)
d7 <- data2 %>% select(-X1)
#############################################################################################
#############################################################################################

# combining all datasets

d8 <- left_join(d1, d2, on = "id")
d9 <- left_join(d8, d3, on = "id")
d10 <- left_join(d9, d4, on = "id")
d11 <- left_join(d10, d5, on = "id")
d12 <- left_join(d11, d6, on = "id")
d13 <- left_join(d12, d7, on = "id")

# if NA for ST call others 

d14 <- d13 %>% mutate(st = replace_na(st, "Other STs"))
d15 <- d14 %>% mutate(pca_cluster = cluster) %>% select(-cluster)
d15 <- d15 %>% mutate(baps_1 = as.factor(baps_1))
d15 <- d15 %>% mutate(pca_cluster = as.factor(pca_cluster))
d15 <- d15 %>% select(id, baps_1, st, cgmlst, kmer_cluster, snp_cluster, PC1, PC2, pca_cluster)
#############################################################################################
# Plot baps1 onto accessory genome

d15$baps_1 <- factor(d15$baps_1, levels=c("1", "2", "3", "4", "5", "6"))
p1 <- ggplot(d15, aes(x = PC1, y = PC2, color = baps_1)) + 
  theme_bw() + 
  xlab("PC1") +
  ylab("PC2") +
  theme(axis.text.y = element_text(size = 40)) +
  theme(axis.title.y = element_text(size = 45, face = "bold")) +
  theme(axis.title.x = element_text(size = 45, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 1, size = 40)) +
  scale_color_manual(values=c("orange", "darkgreen", "darkred", "gray", "darkblue", "darkcyan")) +
   ggtitle("M.") +
  theme(plot.title = element_text(size = 50, face = "bold")) + 
  labs(color='BAPS1') + 
  geom_jitter(pch = 19, height = 1, width = 1, size = 4) +
  theme(legend.title=element_text(size=50, face = "bold")) +
  theme(legend.text=element_text(size=40)) +
  guides(color = guide_legend(override.aes = list(size=20)))
p1

# Plot st onto accessory genome

d15$st <- factor(d15$st, levels=c("ST19", "ST34", "ST36", "ST313", "Other STs"))
p2 <- ggplot(d15, aes(x = PC1, y = PC2, color = st)) + 
  theme_bw() + 
  xlab("PC1") +
  ylab("PC2") +
  theme(axis.text.y = element_text(size = 40)) +
  theme(axis.title.y = element_text(size = 45, face = "bold")) +
  theme(axis.title.x = element_text(size = 45, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 1, size = 40)) +
  scale_color_manual(values=c("darkblue", "darkgreen", "darkred", "gray", "orange")) +
   ggtitle("N.") +
  theme(plot.title = element_text(size = 50, face = "bold")) + 
  labs(color='ST') + 
  geom_jitter(pch = 19, height = 1, width = 1, size = 4) +
  theme(legend.title=element_text(size=50, face = "bold")) +
  theme(legend.text=element_text(size=40)) +
  guides(color = guide_legend(override.aes = list(size=20)))
p2

# Plot cgmlst onto accessory genome

d15$cgmlst <- factor(d15$cgmlst, levels=c("cgMLST 1652656062", "cgMLST 860079270", "Other cgMLSTs"))
p3 <- ggplot(d15, aes(x = PC1, y = PC2, color = cgmlst)) + 
  theme_bw() + 
  xlab("PC1") +
  ylab("PC2") +
  theme(axis.text.y = element_text(size = 40)) +
  theme(axis.title.y = element_text(size = 45, face = "bold")) +
  theme(axis.title.x = element_text(size = 45, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 1, size = 40)) +
  scale_color_manual(values=c("darkblue", "darkred", "gray")) +
   ggtitle("O.") +
  theme(plot.title = element_text(size = 50, face = "bold")) + 
  labs(color='cgMLST') + 
  geom_jitter(pch = 19, height = 1, width = 1, size = 4) +
  theme(legend.title=element_text(size=50, face = "bold")) +
  theme(legend.text=element_text(size=40)) +
  guides(color = guide_legend(override.aes = list(size=20)))
p3

# Plot kmer_cluster onto accessory genome

p4 <- ggplot(d15, aes(x = PC1, y = PC2, color = kmer_cluster)) + 
  theme_bw() + 
  xlab("PC1") +
  ylab("PC2") +
  theme(axis.text.y = element_text(size = 40)) +
  theme(axis.title.y = element_text(size = 45, face = "bold")) +
  theme(axis.title.x = element_text(size = 45, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 1, size = 40)) +
  scale_color_manual(values=c("black", "darkred", "steelblue", "purple", "orange", "darkgreen")) +
   ggtitle("P.") +
  theme(plot.title = element_text(size = 50, face = "bold")) + 
  labs(color='Kmer-clusters') + 
  geom_jitter(pch = 19, height = 1, width = 1, size = 4) +
  theme(legend.title=element_text(size=50, face = "bold")) +
  theme(legend.text=element_text(size=40)) +
  guides(color = guide_legend(override.aes = list(size=20)))
p4

# Plot snp_cluster onto accessory genome 

p5 <- ggplot(d15, aes(x = PC1, y = PC2, color = snp_cluster)) + 
  theme_bw() + 
  xlab("PC1") +
  ylab("PC2") +
  theme(axis.text.y = element_text(size = 40)) +
  theme(axis.title.y = element_text(size = 45, face = "bold")) +
  theme(axis.title.x = element_text(size = 45, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 1, size = 40)) +
  scale_color_manual(values=c("black", "darkred", "steelblue", "purple", "orange", "darkgreen")) +
   ggtitle("Q.") +
  theme(plot.title = element_text(size = 50, face = "bold")) + 
  labs(color='SNP-clusters') + 
  geom_jitter(pch = 19, height = 1, width = 1, size = 4) +
  theme(legend.title=element_text(size=50, face = "bold")) +
  theme(legend.text=element_text(size=40)) +
  guides(color = guide_legend(override.aes = list(size=20)))
p5

# Plot pca_cluster onto accessory genome 

p6 <- ggplot(d15, aes(x = PC1, y = PC2, color = pca_cluster)) + 
  theme_bw() + 
  xlab("PC1") +
  ylab("PC2") +
  theme(axis.text.y = element_text(size = 40)) +
  theme(axis.title.y = element_text(size = 45, face = "bold")) +
  theme(axis.title.x = element_text(size = 45, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 1, size = 40)) +
  scale_color_manual(values=c("gray", "darkred", "darkblue")) +
   ggtitle("R.") +
  theme(plot.title = element_text(size = 50, face = "bold")) + 
  labs(color='PCA-clusters') + 
  geom_jitter(pch = 19, height = 1, width = 1, size = 4) +
  theme(legend.title=element_text(size=50, face = "bold")) +
  theme(legend.text=element_text(size=40)) +
  guides(color = guide_legend(override.aes = list(size=20)))
p6

########################################################################

# retrieving PCA information
a <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/group_1/pca_group_1/wss2.csv')

p7 <- ggplot(a, aes(x = X1, y = wss)) + geom_line() + xlab("Clusters") + ylab("Within cluster sum of squares") +
  scale_x_continuous(breaks = c(seq(from = 0, to = 10, by = 1))) +
  theme(axis.text.y = element_text(size = 40)) +
  theme(axis.title.y = element_text(size = 45, face = "bold")) +
  theme(axis.title.x = element_text(size = 45, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 1, size = 40)) 
p7

# get the variance explained

b <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/group_1/pca_group_1/deviance_model.csv')
b
```






