---
title: "Typhimurium group 10 population structure"
author: "Joao Carlos Gomes-Neto"
date: "3/9/2021"
output: html_document
---


Load packages

```{r, include = FALSE}
# Load packages

library(tidyverse)
library(skimr)
library(RColorBrewer)
library(gridExtra)
library(lattice)
library(ggpubr)
library(vegan)
library(reshape2)
library(ggrepel)
library(ggnewscale)
library(forcats)
library(naniar)
library(data.table)
library(rARPACK)
library(logisticPCA)
library(cluster)
library(factoextra)
library(NbClust)
```

Plots kmer and snp_distance-based distance tSNE analysis Typhimurium 

```{r, include = FALSE}

# Enter kmer-matrix

data1 <- read_csv("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/group_10/kmer_matrix/kmer_group_10_typhimurium.csv")

#####################################################
# Enter MLST results

mlst <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/global_input_data/salmonellast_output.csv')

# QC the data

skim(mlst)

# Generate the id column 

mlst$id2 <- sapply(strsplit(as.character(mlst$FILE),'/'), "[", 3)
s1 <- mlst %>% separate(id2, c("id", "day"), sep = "\\.")

# Select columns of interest

s2 <- s1 %>%
          select(id, ST) %>%
            mutate_all(na_if, "-") %>%
              mutate_all(na_if, "?")

# QC the data

skim(s2)

# st

s2$st <- ifelse(s2$ST == 19, "ST19", 
                   ifelse(s2$ST == 34, "ST34", 
                          ifelse(s2$ST == 36, "ST36",
                                 ifelse(s2$ST == 313, "ST313", "Other STs"))))

# Create data1 

data2 <- s2 

####################################################################

# Enter SISTR results

sistr <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/global_input_data/sistr_all.csv')

# QC the data 

skim(sistr)

# Select columns 

sis1 <- sistr %>%
            select(genome, cgmlst_ST, serovar_cgmlst) %>%
            mutate_all(na_if, "-") %>%
              mutate_all(na_if, "?") %>%
                rename(id = genome)

# Create serovar column

#data1 <- data1 %>% mutate(serovar_cgmlst = replace_na(serovar_cgmlst, 0))
sis1$serovar <- ifelse(sis1$serovar_cgmlst == "Typhimurium", "Biphasic",
                     ifelse(sis1$serovar_cgmlst == "I 4,[5],12:i:-", "Monophasic",
                            "Other serovars"))

# Create data2 and drop rows with missing values

data3 <- sis1 


##########################################################################

# Enter group-Baps

# Enter the data, create a new column for group, and generate the id column
data4 <- read_csv("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/group_10/input_data/roary_output_gr10.fastbaps_l6.csv")
data4$group <- "group_10"
colnames(data4)[1:2] <- c("id", "baps_1")

###########################################################################

# Merge data1 and data2

data5 <- left_join(data1, data2, on = "id")

# Merge data 1, 2, and 3 

data6 <- left_join(data5, data3, on = "id")

# Merge data 1, 2, 3, and 4

data7 <- left_join(data6, data4, on = "id")

# select columns

data8 <- data7 %>%
            select(id, tSNE1, tSNE2, st, cgmlst_ST, serovar, baps_1)

data8 <- mutate(data8, cgmlst = ifelse(cgmlst_ST %in% 1652656062, "cgMLST 1652656062",
                                     ifelse(cgmlst_ST %in% 860079270, "cgMLST 860079270", "Other cgMLSTs")))
########################################################

# Plot serovar 

serovar_data <- data8 %>%
              drop_na(serovar) %>%
              select(tSNE1, tSNE2, serovar) 


sero_plot <- ggplot(serovar_data, aes(x = tSNE1, y = tSNE2, color = serovar)) + 
  theme_bw() + 
  xlab("tSNE1") +
  ylab("tSNE2") +
  theme(axis.text.y = element_text(size = 40)) +
  theme(axis.title.y = element_text(size = 45, face = "bold")) +
  theme(axis.title.x = element_text(size = 45, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 1, size = 40)) +
  scale_color_manual(values=c("gray", "darkblue", "red")) +
   ggtitle("") +
  theme(plot.title = element_text(size = 50, face = "bold")) + 
  labs(color='Serovar') + 
  geom_jitter(pch = 21, height = 1, width = 1, size = 4) +
  theme(legend.title=element_text(size=50, face = "bold")) +
  theme(legend.text=element_text(size=40))
sero_plot

# Plot baps_1

baps_data <- data8 %>%
              drop_na(baps_1) %>%
              select(tSNE1, tSNE2, baps_1) 
baps_data$baps_1 <- factor(baps_data$baps_1, levels=c("1", "2", "3", "4", "5", "6"))


baps_plot <- ggplot(baps_data, aes(x = tSNE1, y = tSNE2, color = baps_1)) + 
  theme_bw() + 
  xlab("tSNE1") +
  ylab("tSNE2") +
  theme(axis.text.y = element_text(size = 40)) +
  theme(axis.title.y = element_text(size = 45, face = "bold")) +
  theme(axis.title.x = element_text(size = 45, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 1, size = 40)) +
  scale_color_manual(values=c("orange", "darkgreen", "darkred", "gray", "darkblue", "darkcyan")) +
   ggtitle("A.") +
  theme(plot.title = element_text(size = 50, face = "bold")) + 
  labs(color='BAPS1') + 
  geom_jitter(pch = 19, height = 1, width = 1, size = 4) +
  theme(legend.title=element_text(size=50, face = "bold")) +
  theme(legend.text=element_text(size=40)) +
      guides(color = guide_legend(override.aes = list(size=20)))
baps_plot


# Plot ST

st_data <- data8 %>%
              #drop_na(st) %>%
  mutate(st = replace_na(st, "Other STs")) %>%
              select(tSNE1, tSNE2, st) 
st_data$st <- factor(st_data$st, levels=c("ST19", "ST34", "ST36", "ST313", "Other STs"))


st_plot <- ggplot(st_data, aes(x = tSNE1, y = tSNE2, color = st)) + 
  theme_bw() + 
  xlab("tSNE1") +
  ylab("tSNE2") +
  theme(axis.text.y = element_text(size = 40)) +
  theme(axis.title.y = element_text(size = 45, face = "bold")) +
  theme(axis.title.x = element_text(size = 45, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 1, size = 40)) +
  scale_color_manual(values=c("darkblue", "darkgreen", "darkred", "gray", "orange")) +
   ggtitle("B.") +
  theme(plot.title = element_text(size = 50, face = "bold")) + 
  labs(color='ST') + 
  geom_jitter(pch = 19, height = 1, width = 1, size = 4) +
  theme(legend.title=element_text(size=50, face = "bold")) +
  theme(legend.text=element_text(size=40)) +
      guides(color = guide_legend(override.aes = list(size=20)))
st_plot


# Plot cgmlst 

cgmlst_data <- data8 %>%
              drop_na(cgmlst) %>%
              select(tSNE1, tSNE2, cgmlst) 
cgmlst_data$cgmlst <- factor(cgmlst_data$cgmlst, levels=c("cgMLST 1652656062", "cgMLST 860079270", "Other cgMLSTs"))


cgmlst_plot <- ggplot(cgmlst_data, aes(x = tSNE1, y = tSNE2, color = cgmlst)) + 
  theme_bw() + 
  xlab("tSNE1") +
  ylab("tSNE2") +
  theme(axis.text.y = element_text(size = 40)) +
  theme(axis.title.y = element_text(size = 45, face = "bold")) +
  theme(axis.title.x = element_text(size = 45, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 1, size = 40)) +
  scale_color_manual(values=c("darkblue", "darkred", "gray")) +
   ggtitle("C.") +
  theme(plot.title = element_text(size = 50, face = "bold")) + 
  labs(color='cgMLST') + 
  geom_jitter(pch = 19, height = 1, width = 1, size = 4) +
  theme(legend.title=element_text(size=50, face = "bold")) +
  theme(legend.text=element_text(size=40)) +
      guides(color = guide_legend(override.aes = list(size=20)))
cgmlst_plot

##################################################################################

# Enter snp-matrix

data9 <- read_csv("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/group_10/snp_matrix/snp_group_10_typhimurium.csv")


data10 <- data8[, -c(2:3)]

# Merge data 9 and 10

data11 <- left_join(data9, data10, on = "id")

# Plot serovar 

serovar_data2 <- data11 %>%
              drop_na(serovar) %>%
              select(tSNE1, tSNE2, serovar) 


sero_plot2 <- ggplot(serovar_data2, aes(x = tSNE1, y = tSNE2, color = serovar)) + 
  theme_bw() + 
  xlab("tSNE1") +
  ylab("tSNE2") +
  theme(axis.text.y = element_text(size = 40)) +
  theme(axis.title.y = element_text(size = 45, face = "bold")) +
  theme(axis.title.x = element_text(size = 45, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 1, size = 40)) +
  scale_color_manual(values=c("gray", "darkblue", "red")) +
   ggtitle("") +
  theme(plot.title = element_text(size = 50, face = "bold")) + 
  labs(color='Serovar') + 
  geom_jitter(pch = 21, height = 1, width = 1, size = 4) +
  theme(legend.title=element_text(size=50, face = "bold")) +
  theme(legend.text=element_text(size=40))
sero_plot2

# Plot baps_1

baps_data2 <- data11 %>%
              drop_na(baps_1) %>%
              select(tSNE1, tSNE2, baps_1) 
baps_data2$baps_1 <- factor(baps_data2$baps_1, levels=c("1", "2", "3", "4", "5", "6"))


baps_plot2 <- ggplot(baps_data2, aes(x = tSNE1, y = tSNE2, color = baps_1)) + 
  theme_bw() + 
  xlab("tSNE1") +
  ylab("tSNE2") +
  theme(axis.text.y = element_text(size = 40)) +
  theme(axis.title.y = element_text(size = 45, face = "bold")) +
  theme(axis.title.x = element_text(size = 45, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 1, size = 40)) +
  scale_color_manual(values=c("orange", "darkgreen", "darkred", "gray", "darkblue", "darkcyan")) +
   ggtitle("E.") +
  theme(plot.title = element_text(size = 50, face = "bold")) + 
  labs(color='BAPS1') + 
  geom_jitter(pch = 19, height = 1, width = 1, size = 4) +
  theme(legend.title=element_text(size=50, face = "bold")) +
  theme(legend.text=element_text(size=40)) +
      guides(color = guide_legend(override.aes = list(size=20)))
baps_plot2


# Plot ST

st_data2 <- data11 %>%
              #drop_na(st) %>%
  mutate(st = replace_na(st, "Other STs")) %>%
              select(tSNE1, tSNE2, st) 
st_data2$st <- factor(st_data2$st, levels=c("ST19", "ST34", "ST36", "ST313", "Other STs"))


st_plot2 <- ggplot(st_data2, aes(x = tSNE1, y = tSNE2, color = st)) + 
  theme_bw() + 
  xlab("tSNE1") +
  ylab("tSNE2") +
  theme(axis.text.y = element_text(size = 40)) +
  theme(axis.title.y = element_text(size = 45, face = "bold")) +
  theme(axis.title.x = element_text(size = 45, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 1, size = 40)) +
  scale_color_manual(values=c("darkblue", "darkgreen", "darkred", "gray", "orange")) +
   ggtitle("F.") +
  theme(plot.title = element_text(size = 50, face = "bold")) + 
  labs(color='ST') + 
  geom_jitter(pch = 19, height = 1, width = 1, size = 4) +
  theme(legend.title=element_text(size=50, face = "bold")) +
  theme(legend.text=element_text(size=40)) +
  guides(color = guide_legend(override.aes = list(size=20)))
st_plot2


# Plot cgmlst 

cgmlst_data2 <- data11 %>%
              drop_na(cgmlst) %>%
              select(tSNE1, tSNE2, cgmlst) 
cgmlst_data2$cgmlst <- factor(cgmlst_data2$cgmlst, levels=c("cgMLST 1652656062", "cgMLST 860079270", "Other cgMLSTs"))


cgmlst_plot2 <- ggplot(cgmlst_data2, aes(x = tSNE1, y = tSNE2, color = cgmlst)) + 
  theme_bw() + 
  xlab("tSNE1") +
  ylab("tSNE2") +
  theme(axis.text.y = element_text(size = 40)) +
  theme(axis.title.y = element_text(size = 45, face = "bold")) +
  theme(axis.title.x = element_text(size = 45, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 1, size = 40)) +
  scale_color_manual(values=c("darkblue", "darkred", "gray")) +
   ggtitle("G.") +
  theme(plot.title = element_text(size = 50, face = "bold")) + 
  labs(color='cgMLST') + 
  geom_jitter(pch = 19, height = 1, width = 1, size = 4) +
  theme(legend.title=element_text(size=50, face = "bold")) +
  theme(legend.text=element_text(size=40)) +
  guides(color = guide_legend(override.aes = list(size=20)))
cgmlst_plot2
```

Cluster analysis using core genome information Typhimurium

```{r, include = FALSE}

# kmer data (2d)
# enter the data

data1 <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/group_10/kmer_matrix/kmer_group_10_typhimurium.csv')

#  Non-supervised kmeans clustering analysis

cluster_data <- data1 %>% select(tSNE1, tSNE2)

# determine the number of clusters with the elbow method

set.seed(123)

# function to compute total within-cluster sum of square 
wss <- function(k) {
  kmeans(cluster_data, k, nstart = 10 )$tot.withinss
}

# Compute and plot wss for k = 1 to k = 15
k.values <- 1:15

# extract wss for 2-15 clusters
wss_values <- map_dbl(k.values, wss)

plot(k.values, wss_values,
       type="b", pch = 19, frame = FALSE, 
       xlab="Number of clusters K",
       ylab="Total within-clusters sum of squares")

# or can use this function
set.seed(123)

p1 <- fviz_nbclust(cluster_data, kmeans, method = "wss")
p1 

# silhouette method
set.seed(123)
a1 <- fviz_nbclust(cluster_data, kmeans, method = "silhouette")
a1 

# combining both approaches we determine that 2 clusters are enough 
set.seed(123)
km.final <- kmeans(cluster_data, 6)
data1$cluster <- km.final$cluster
data1$cluster <- as.factor(data1$cluster)

# plot the kmer 2d cluster data

kmercluster <- ggplot(data1, aes(x = tSNE1, y = tSNE2, color = cluster)) + 
  theme_bw() + 
  xlab("tSNE1") +
  ylab("tSNE2") +
  theme(axis.text.y = element_text(size = 40)) +
  theme(axis.title.y = element_text(size = 45, face = "bold")) +
  theme(axis.title.x = element_text(size = 45, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 1, size = 40)) +
  scale_color_manual(values=c("black", "darkred", "steelblue", "purple", "orange", "darkgreen")) +
   ggtitle("D.") +
  theme(plot.title = element_text(size = 50, face = "bold")) + 
  labs(color='Kmeans clusters') + 
  geom_jitter(pch = 19, height = 1, width = 1, size = 4) +
  theme(legend.title=element_text(size=50, face = "bold")) +
  theme(legend.text=element_text(size=40)) +
  guides(color = guide_legend(override.aes = list(size=20)))
kmercluster

######################################################################################

# snp data (2d)
# enter the data

data1 <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/group_10/snp_matrix/snp_group_10_typhimurium.csv')

#  Non-supervised kmeans clustering analysis

cluster_data <- data1 %>% select(tSNE1, tSNE2)

# determine the number of clusters with the elbow method

set.seed(123)

# function to compute total within-cluster sum of square 
wss <- function(k) {
  kmeans(cluster_data, k, nstart = 10 )$tot.withinss
}

# Compute and plot wss for k = 1 to k = 15
k.values <- 1:15

# extract wss for 2-15 clusters
wss_values <- map_dbl(k.values, wss)

plot(k.values, wss_values,
       type="b", pch = 19, frame = FALSE, 
       xlab="Number of clusters K",
       ylab="Total within-clusters sum of squares")

# or can use this function
set.seed(123)

p2 <- fviz_nbclust(cluster_data, kmeans, method = "wss")
p2 

# silhouette method
set.seed(123)
a2 <- fviz_nbclust(cluster_data, kmeans, method = "silhouette")
a2 

# combining both approaches we determine that 2 clusters are enough 
set.seed(123)
km.final <- kmeans(cluster_data, 6)
data1$cluster <- km.final$cluster
data1$cluster <- as.factor(data1$cluster)

# plot the kmer 2d cluster data

snpcluster <- ggplot(data1, aes(x = tSNE1, y = tSNE2, color = cluster)) + 
  theme_bw() + 
  xlab("tSNE1") +
  ylab("tSNE2") +
  theme(axis.text.y = element_text(size = 40)) +
  theme(axis.title.y = element_text(size = 45, face = "bold")) +
  theme(axis.title.x = element_text(size = 45, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 1, size = 40)) +
  scale_color_manual(values=c("black", "darkred", "steelblue", "purple", "orange", "darkgreen")) +
   ggtitle("H.") +
  theme(plot.title = element_text(size = 50, face = "bold")) + 
  labs(color='Kmeans clusters') + 
  geom_jitter(pch = 19, height = 1, width = 1, size = 4) +
  theme(legend.title=element_text(size=50, face = "bold")) +
  theme(legend.text=element_text(size=40)) +
 guides(color = guide_legend(override.aes = list(size=20))) 
snpcluster
```

Salmonella Typhimurium plots using shell loci

```{r, include = FALSE}

# Enter group1-Baps

# Enter the data, create a new column for group, and generate the id column
data4 <- read_csv("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/group_10/input_data/roary_output_gr10.fastbaps_l6.csv")
data4$group <- "group_10"
colnames(data4)[1:2] <- c("id", "baps_1")

d1 <- data4 %>% select(id, baps_1, group)
###########################################################################
# Enter MLST results

mlst <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/global_input_data/salmonellast_output.csv')

# QC the data

skim(mlst)

# Generate the id column 

mlst$id2 <- sapply(strsplit(as.character(mlst$FILE),'/'), "[", 3)
s1 <- mlst %>% separate(id2, c("id", "day"), sep = "\\.")

# Select columns of interest

s2 <- s1 %>%
          select(id, ST) %>%
            mutate_all(na_if, "-") %>%
              mutate_all(na_if, "?")

# QC the data

skim(s2)

# st

s2$st <- ifelse(s2$ST == 19, "ST19", 
                   ifelse(s2$ST == 34, "ST34", 
                          ifelse(s2$ST == 36, "ST36",
                                 ifelse(s2$ST == 313, "ST313", "Other STs"))))

# Create data1 

d2 <- s2 

####################################################################

# Enter SISTR results

sistr <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/global_input_data/sistr_all.csv')

# QC the data 

skim(sistr)

# Select columns 

sis1 <- sistr %>%
            select(genome, cgmlst_ST, serovar_cgmlst) %>%
            mutate_all(na_if, "-") %>%
              mutate_all(na_if, "?") %>%
                rename(id = genome)

d3 <- sis1 

d3 <- mutate(d3, cgmlst = ifelse(cgmlst_ST %in% 1652656062, "cgMLST 1652656062",
                                     ifelse(cgmlst_ST %in% 860079270, "cgMLST 860079270", "Other cgMLSTs")))
##########################################################################
# KMER

data1 <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/group_10/kmer_matrix/kmer_group_10_typhimurium.csv')

#  Non-supervised kmeans clustering analysis

cluster_data <- data1 %>% select(tSNE1, tSNE2)

# combining both approaches we determine that 2 clusters are enough 
set.seed(123)
km.final <- kmeans(cluster_data, 6)
data1$kmer_cluster <- km.final$cluster
data1$kmer_cluster <- as.factor(data1$kmer_cluster)

# select the column we need

d4 <- data1 %>% select(id, kmer_cluster)
##################################################################################

# SNP

data1 <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/group_10/snp_matrix/snp_group_10_typhimurium.csv')

#  Non-supervised kmeans clustering analysis

cluster_data <- data1 %>% select(tSNE1, tSNE2)

# combining both approaches we determine that 2 clusters are enough 
set.seed(123)
km.final <- kmeans(cluster_data, 6)
data1$snp_cluster <- km.final$cluster
data1$snp_cluster <- as.factor(data1$snp_cluster)

# filter the column

d5 <- data1 %>% select(id, snp_cluster)

####################################################################################
#############################################################################################

# PCA CLUSTER

data1 <- read_csv("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/group_10/pca_group_10/pca_data2.csv")
data2 <- read_csv("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/group_10/pca_group_10/cluster2.csv")

d6 <- data1 %>% select(-X1)
d7 <- data2 %>% select(-X1)
#############################################################################################
#############################################################################################

# combining all datasets

d8 <- left_join(d1, d2, on = "id")
d9 <- left_join(d8, d3, on = "id")
d10 <- left_join(d9, d4, on = "id")
d11 <- left_join(d10, d5, on = "id")
d12 <- left_join(d11, d6, on = "id")
d13 <- left_join(d12, d7, on = "id")

# if NA for ST call others 

d14 <- d13 %>% mutate(st = replace_na(st, "Other STs"))
d15 <- d14 %>% mutate(pca_cluster = cluster) %>% select(-cluster)
d15 <- d15 %>% mutate(baps_1 = as.factor(baps_1))
d15 <- d15 %>% mutate(pca_cluster = as.factor(pca_cluster))
d15 <- d15 %>% select(id, baps_1, st, cgmlst, kmer_cluster, snp_cluster, PC1, PC2, pca_cluster)
#############################################################################################
# Plot baps1 onto accessory genome

d15$baps_1 <- factor(d15$baps_1, levels=c("1", "2", "3", "4", "5", "6"))
p1 <- ggplot(d15, aes(x = PC1, y = PC2, color = baps_1)) + 
  theme_bw() + 
  xlab("PC1") +
  ylab("PC2") +
  theme(axis.text.y = element_text(size = 40)) +
  theme(axis.title.y = element_text(size = 45, face = "bold")) +
  theme(axis.title.x = element_text(size = 45, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 1, size = 40)) +
  scale_color_manual(values=c("orange", "darkgreen", "darkred", "gray", "darkblue", "darkcyan")) +
   ggtitle("I.") +
  theme(plot.title = element_text(size = 50, face = "bold")) + 
  labs(color='BAPS1') + 
  geom_jitter(pch = 19, height = 1, width = 1, size = 4) +
  theme(legend.title=element_text(size=50, face = "bold")) +
  theme(legend.text=element_text(size=40)) +
  guides(color = guide_legend(override.aes = list(size=20)))
p1

# Plot st onto accessory genome

d15$st <- factor(d15$st, levels=c("ST19", "ST34", "ST36", "ST313", "Other STs"))
p2 <- ggplot(d15, aes(x = PC1, y = PC2, color = st)) + 
  theme_bw() + 
  xlab("PC1") +
  ylab("PC2") +
  theme(axis.text.y = element_text(size = 40)) +
  theme(axis.title.y = element_text(size = 45, face = "bold")) +
  theme(axis.title.x = element_text(size = 45, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 1, size = 40)) +
  scale_color_manual(values=c("darkblue", "darkgreen", "darkred", "gray", "orange")) +
   ggtitle("J.") +
  theme(plot.title = element_text(size = 50, face = "bold")) + 
  labs(color='ST') + 
  geom_jitter(pch = 19, height = 1, width = 1, size = 4) +
  theme(legend.title=element_text(size=50, face = "bold")) +
  theme(legend.text=element_text(size=40)) +
  guides(color = guide_legend(override.aes = list(size=20)))
p2

# Plot cgmlst onto accessory genome

d15$cgmlst <- factor(d15$cgmlst, levels=c("cgMLST 1652656062", "cgMLST 860079270", "Other cgMLSTs"))
p3 <- ggplot(d15, aes(x = PC1, y = PC2, color = cgmlst)) + 
  theme_bw() + 
  xlab("PC1") +
  ylab("PC2") +
  theme(axis.text.y = element_text(size = 40)) +
  theme(axis.title.y = element_text(size = 45, face = "bold")) +
  theme(axis.title.x = element_text(size = 45, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 1, size = 40)) +
  scale_color_manual(values=c("darkblue", "darkred", "gray")) +
   ggtitle("K.") +
  theme(plot.title = element_text(size = 50, face = "bold")) + 
  labs(color='cgMLST') + 
  geom_jitter(pch = 19, height = 1, width = 1, size = 4) +
  theme(legend.title=element_text(size=50, face = "bold")) +
  theme(legend.text=element_text(size=40)) +
  guides(color = guide_legend(override.aes = list(size=20)))
p3

# Plot kmer_cluster onto accessory genome

p4 <- ggplot(d15, aes(x = PC1, y = PC2, color = kmer_cluster)) + 
  theme_bw() + 
  xlab("PC1") +
  ylab("PC2") +
  theme(axis.text.y = element_text(size = 40)) +
  theme(axis.title.y = element_text(size = 45, face = "bold")) +
  theme(axis.title.x = element_text(size = 45, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 1, size = 40)) +
  scale_color_manual(values=c("black", "darkred", "steelblue", "purple", "orange", "darkgreen")) +
   ggtitle("L.") +
  theme(plot.title = element_text(size = 50, face = "bold")) + 
  labs(color='Kmer-clusters') + 
  geom_jitter(pch = 19, height = 1, width = 1, size = 4) +
  theme(legend.title=element_text(size=50, face = "bold")) +
  theme(legend.text=element_text(size=40)) +
  guides(color = guide_legend(override.aes = list(size=20)))
p4

# Plot snp_cluster onto accessory genome 

p5 <- ggplot(d15, aes(x = PC1, y = PC2, color = snp_cluster)) + 
  theme_bw() + 
  xlab("PC1") +
  ylab("PC2") +
  theme(axis.text.y = element_text(size = 40)) +
  theme(axis.title.y = element_text(size = 45, face = "bold")) +
  theme(axis.title.x = element_text(size = 45, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 1, size = 40)) +
  scale_color_manual(values=c("black", "darkred", "steelblue", "purple", "orange", "darkgreen")) +
   ggtitle("M.") +
  theme(plot.title = element_text(size = 50, face = "bold")) + 
  labs(color='SNP-clusters') + 
  geom_jitter(pch = 19, height = 1, width = 1, size = 4) +
  theme(legend.title=element_text(size=50, face = "bold")) +
  theme(legend.text=element_text(size=40)) +
  guides(color = guide_legend(override.aes = list(size=20)))
p5

# Plot pca_cluster onto accessory genome 

p6 <- ggplot(d15, aes(x = PC1, y = PC2, color = pca_cluster)) + 
  theme_bw() + 
  xlab("PC1") +
  ylab("PC2") +
  theme(axis.text.y = element_text(size = 40)) +
  theme(axis.title.y = element_text(size = 45, face = "bold")) +
  theme(axis.title.x = element_text(size = 45, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 1, size = 40)) +
  scale_color_manual(values=c("gray", "darkred", "darkblue")) +
   ggtitle("N.") +
  theme(plot.title = element_text(size = 50, face = "bold")) + 
  labs(color='PCA-clusters') + 
  geom_jitter(pch = 19, height = 1, width = 1, size = 4) +
  theme(legend.title=element_text(size=50, face = "bold")) +
  theme(legend.text=element_text(size=40)) +
  guides(color = guide_legend(override.aes = list(size=20)))
p6

########################################################################

# retrieving PCA information
a <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/group_10/pca_group_10/wss2.csv')

p7 <- ggplot(a, aes(x = X1, y = wss)) + geom_line() + xlab("Clusters") + ylab("Within cluster sum of squares") +
  scale_x_continuous(breaks = c(seq(from = 0, to = 10, by = 1))) +
  theme(axis.text.y = element_text(size = 40)) +
  theme(axis.title.y = element_text(size = 45, face = "bold")) +
  theme(axis.title.x = element_text(size = 45, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 1, size = 40)) +
  ggtitle("O.") +
  theme(plot.title = element_text(size = 50, face = "bold")) 
p7

# get the variance explained

b <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/group_10/pca_group_10/deviance_model.csv')
b
```