---
title: "Figure 5 Monophasic data pangenome analysis"
author: "Joao Carlos Gomes-Neto"
date: "2/25/2021"
output: html_document
---

Load packages

```{r, include = FALSE}
# Load packages

library(tidyverse)
library(skimr)
library(RColorBrewer)
library(gridExtra)
library(lattice)
library(ggpubr)
library(vegan)
library(reshape2)
library(ggrepel)
library(ggnewscale)
library(forcats)
library(naniar)
library(data.table)
library(rARPACK)
library(logisticPCA)
library(cluster)
library(factoextra)
library(NbClust)
library(plotly)
```

Generating metadata for Salmonella Typhimurium heatmap plot

```{r, include = FALSE}

# Enter group1-Baps

# Enter the data, create a new column for group, and generate the id column
data4 <- read_csv("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/group_1/input_data/roary_output_gr1.fastbaps_l6.csv")
data4$group <- "group_1"
colnames(data4)[1:2] <- c("id", "baps_1")

d1 <- data4 %>% select(id, baps_1, group)
###########################################################################
# Enter MLST results

mlst <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/global_input_data/salmonellast_output.csv')

# QC the data

skim(mlst)

# Generate the id column 

mlst$id2 <- sapply(strsplit(as.character(mlst$FILE),'/'), "[", 3)
s1 <- mlst %>% separate(id2, c("id", "day"), sep = "\\.")

# Select columns of interest

s2 <- s1 %>%
          select(id, ST) %>%
            mutate_all(na_if, "-") %>%
              mutate_all(na_if, "?")

# QC the data

skim(s2)

# st

s2$st <- ifelse(s2$ST == 19, "ST19", 
                   ifelse(s2$ST == 34, "ST34", 
                          ifelse(s2$ST == 36, "ST36",
                                 ifelse(s2$ST == 313, "ST313", "Other STs"))))

# Create data1 

d2 <- s2 

####################################################################

# Enter SISTR results

sistr <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/global_input_data/sistr_all.csv')

# QC the data 

skim(sistr)

# Select columns 

sis1 <- sistr %>%
            select(genome, cgmlst_ST, serovar_cgmlst) %>%
            mutate_all(na_if, "-") %>%
              mutate_all(na_if, "?") %>%
                rename(id = genome)

d3 <- sis1 

d3 <- mutate(d3, cgmlst = ifelse(cgmlst_ST %in% 1652656062, "cgMLST 1652656062",
                                     ifelse(cgmlst_ST %in% 860079270, "cgMLST 860079270", "Other cgMLSTs")))
##########################################################################
# KMER

data1 <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/group_1/kmer_matrix/kmer_group_1_typhimurium.csv')

#  Non-supervised kmeans clustering analysis

cluster_data <- data1 %>% select(tSNE1, tSNE2)

# combining both approaches we determine that 2 clusters are enough 
set.seed(123)
km.final <- kmeans(cluster_data, 6)
data1$kmer_cluster <- km.final$cluster
data1$kmer_cluster <- as.factor(data1$kmer_cluster)

# select the column we need

d4 <- data1 %>% select(id, kmer_cluster)
##################################################################################

# SNP

data1 <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/group_1/snp_matrix/snp_group_1_typhimurium.csv')

#  Non-supervised kmeans clustering analysis

cluster_data <- data1 %>% select(tSNE1, tSNE2)

# combining both approaches we determine that 2 clusters are enough 
set.seed(123)
km.final <- kmeans(cluster_data, 6)
data1$snp_cluster <- km.final$cluster
data1$snp_cluster <- as.factor(data1$snp_cluster)

# filter the column

d5 <- data1 %>% select(id, snp_cluster)

####################################################################################
#############################################################################################

# PCA CLUSTER

data1 <- read_csv("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/group_1/pca_group_1/cluster2.csv")

d6 <- data1 %>% select(-X1)
#############################################################################################
#############################################################################################

# combining all datasets

d8 <- left_join(d1, d2, on = "id")
d9 <- left_join(d8, d3, on = "id")
d10 <- left_join(d9, d4, on = "id")
d11 <- left_join(d10, d5, on = "id")
d12 <- left_join(d11, d6, on = "id")
d13 <- left_join(d12, d7, on = "id")

# if NA for ST call others 

d14 <- d13 %>% mutate(st = replace_na(st, "Other STs"))
d15 <- d14 %>% mutate(pca_cluster = cluster) %>% select(-cluster)
d15 <- d15 %>% mutate(baps_1 = as.factor(baps_1))
d15 <- d15 %>% mutate(pca_cluster = as.factor(pca_cluster))
d15 <- d15 %>% select(id, baps_1, st, cgmlst, kmer_cluster, snp_cluster, pca_cluster)

#############################################################################################
#############################################################################################

# Getting the AMR data 

abx1 <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/global_input_data/sabricate_resfinder_output.csv')
abx1$id2 <- sapply(strsplit(as.character(abx1$'#FILE'),'/'), "[", 3)
abx1 <- abx1 %>% separate(id2, c("id", "garbage"), sep = "\\.")
abx1 <- abx1 %>% mutate(gene = GENE)
abx2 <- abx1 %>% select(id, gene)

# eliminate genes with double or more counts

abx3 <- abx2 %>% group_by(id, gene) %>% count()
abx4 <- abx3 %>% filter(n <= 1)

# spread the data

abx5 <- spread(abx4, key = gene, value = n)

# input zeros for replacing NAs

abx5[is.na(abx5)] <- 0

# merge with d15

d16 <- left_join(d15, abx5, on = "id")
d17 <- column_to_rownames(d16, var = "id")

# export file as csv

group_1_abx <- d17
write.csv(group_1_abx, "~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/group_1/input_data/group_1_abx.csv")

# filter genes that we need

d18 <- group_1_abx %>% select(baps_1, st, cgmlst, kmer_cluster, snp_cluster, pca_cluster,
                                             `aac(3)-Ib_1`, `aac(6')-Iaa_1`, aadA2_1, `ant(3'')-Ia_1`, `aph(3'')-Ib_5`,
                                             `aph(6)-Id_1`, `blaCARB-2_1`, `blaTEM-1B_1`, catA1_1,
                                             dfrA1_10, floR_2, sul1_5, sul1_10, sul2_2, sul2_3, sul2_6,
                                             `tet(A)_6`, `tet(B)_2`, `tet(G)_2`, `mdf(A)_1`)

d19 <- rownames_to_column(d18, var = "id")
#############################################################################################
#############################################################################################

# Getting the plasmid data 

x1 <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/global_input_data/sabricate_plasmidfinder_output.csv')
x1$id2 <- sapply(strsplit(as.character(x1$'#FILE'),'/'), "[", 3)
x1 <- x1 %>% separate(id2, c("id", "garbage"), sep = "\\.")
x1 <- x1 %>% mutate(gene = GENE)
x2 <- x1 %>% select(id, gene)

# eliminate genes with double or more counts

x3 <- x2 %>% group_by(id, gene) %>% count()
x4 <- x3 %>% filter(n <= 1)

# spread the data

x5 <- spread(x4, key = gene, value = n)

# input zeros for replacing NAs

x5[is.na(x5)] <- 0

# merge with d15

d16 <- left_join(d15, x5, on = "id")
d17 <- column_to_rownames(d16, var = "id")
d17[is.na(d17)] <- 0

# export file as csv

group_1_plasmid <- d17
write.csv(group_1_plasmid, "~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/group_1/input_data/group_1_plasmid.csv")

# filter plasmids that we need

d18b <- group_1_plasmid %>% select(ColRNAI_1, `IncA/C2_1`, `IncFIB(S)_1`, `IncFII(S)_1`, IncQ1_1)

d19b <- rownames_to_column(d18b, var = "id")
######################################################################################
######################################################################################
######################################################################################

# combine AMR and plasmids into one dataset

d20 <- left_join(d19, d19b, on = "id")
#d20 <- column_to_rownames(d20, var = "id")

#####################################################################################
#####################################################################################
# PANGEA results
# Enter empirical results

data1 <- read_csv("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/gwas_genes/typhimurium_st34_gene_gwas_filtered.csv")

# Find top predictors with 70% coverage and above

data2 <- data1%>% 
  group_by(gene_kmer_id) %>% 
  summarise(n = n()) %>%
  filter(n >= 10)

# get the list of genes

list_genes <- data2$gene_kmer_id

# Subset the original data based on that
data3 <- data1 %>%
      filter(gene_kmer_id %in% list_genes) 

# Create a column called empirical 

data3$results <- "Empirical"

####################################################

# Enter random results 

data4 <- read_csv("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/gwas_genes/random_typhimurium_st34_gene_gwas_not_filtered.csv")

# Subset the original data based on that
data5 <- data4 %>%
      filter(gene_kmer_id %in% list_genes) 

# Create a column called empirical 

data5$results <- "Random"

##################################################

# Merge empirical and random results

data6 <- rbind(data3, data5)

data6$results <- factor(data6$results, levels=c("Random", "Empirical"))
data7 <- data6 %>% select(results, gene_kmer_id, accur, neg_pred_value, pos_pred_value)
data7 <- data7 %>% group_by(results, gene_kmer_id) %>% summarize(mean_accur = mean(accur),
                                                                 mean_pos = mean(pos_pred_value),
                                                                 mean_neg = mean(neg_pred_value))
data7 <- rename(data7, Accuracy = mean_accur)
# Plot data

p9b <- ggplot(data7, aes(x = mean_pos, y = gene_kmer_id, size = mean_neg, color = Accuracy)) +
  theme_bw() +
  xlab("Positive predictive value") +
  ylab("") +
  xlim(0, 1) +
  theme(axis.text.y = element_text(size = 30)) +
  theme(axis.title.y = element_text(size = 40, face = "bold")) +
  theme(axis.title.x = element_text(size = 40, face = "bold")) +
  theme(axis.text.x = element_text(size = 30, angle = 45, vjust = 1, hjust=1)) +
  theme(strip.text.x = element_text(size = 35, colour = "black", angle = 0)) +
  geom_point() + 
  ggtitle("ST34 vs Other STs (loci)") +  
  theme(plot.title = element_text(size = 50, face = "bold")) + 
  theme(legend.title=element_text(size=40, face = "bold")) +
  theme(legend.text=element_text(size=23)) +
  scale_size_continuous(name = "Negative predictive value") +
  scale_colour_gradient(low = "orange", high = "darkblue") +
  facet_wrap(~results)
p9b

################################################################################
################################################################################
################################################################################

# enter Rtab file and select genes needed

d1 <- read_tsv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/group_1/input_data/gene_presence_absence.Rtab')

# set Genes as rownames

d1 <- column_to_rownames(d1, var = "Gene") 

d1$Gene <- NULL

# Transpose it

d2 <- transpose(d1)

# Add column and rownames

rownames(d2) <- colnames(d1)
colnames(d2) <- rownames(d1)

# Reset id to colnames

d3 <- rownames_to_column(d2, "id")

# List of targeted genes 

# metal resistance - SGI3-4

metals <- c("pcoA", "pcoB", "pcoC", "pcoD", "pcoE", "pcoG", "pcoR", "pcoS", "pco",
            "silA", "silB", "silC", "silE", "silP", "silR", "silS", "sil",
            "cusA", "cusB", "cusC", "cusF", "cusR", "cusS", "cus",
            "arsA", "arsB", "arsC", "arsD", "arsR", "ars",
            "merA", "merC", "merP", "merR", "merT",
            "zitB", 
            "zntA", "zntB", "zntR")

######################################################################

d4 <- d3 %>% select("id", contains(metals))

remove_genes <- c("cusA_2", "cusB_2", "cusC_1", "cusC_2", "dcuS", "dcuS_1", "cusS_2", "arsA",
                  "arsC_1", "arsC_2", "arsC_3", "arsD", "arsR2", "arsH", "merR1", "merT_2", "merT_1", 
                  "zitB_1", "zntA_1", "zntB_1", "zntB_2", "zntR_2", "zntR_1")
d5 <- d4 %>% select(!remove_genes)

# merge with d20

d21 <- left_join(d20, d5, on = "id")
lociPopStr_st34 <- column_to_rownames(d21, var = "id")
write.csv(lociPopStr_st34, "~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/group_1/input_data/lociPopStr_st34.csv")
```

Analysis of samples containing clinical bovine and human isolates

```{r, include = FALSE}

# Enter group1-Baps

# Enter the data, create a new column for group, and generate the id column
data4 <- read_csv("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/unmc_genomes/fastbaps_baps.csv")
colnames(data4)[1:2] <- c("id", "baps_1")

d1 <- data4 %>% select(id, baps_1)
###########################################################################
# Enter MLST results

mlst <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/unmc_genomes/mlst_output.csv')

# Generate the id column 

mlst$id <- sapply(strsplit(as.character(mlst$FILE),'_'), "[", 1)

# Create data1 

d2 <- mlst %>% select(id, ST)
d2 <- d2 %>% mutate(st = case_when(ST == 19 ~ "ST19",
                                           ST == 34 ~ "ST34",
                                           ST == 2072 ~ "ST2072",
                                           ST == 2379 ~ "ST2379"))

d2 <- d2 %>% select(id, st)
####################################################################

# Enter SISTR results

sistr <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/unmc_genomes/sistr_all_merge.csv')

d3 <- sistr %>% select(genome, cgmlst_ST) %>% rename(id2 = genome)
d3$id <- sapply(strsplit(as.character(d3$id2),'_'), "[", 1)
d3 <- d3 %>% select(id, cgmlst_ST)
d3[is.na(d3)] <- 0
d3 <- d3 %>% mutate(cgmlst = case_when(cgmlst_ST == 49167135 ~ "cgMLST 49167135",
                                           cgmlst_ST == 497195620 ~ "cgMLST 497195620",
                                           cgmlst_ST == 499616518 ~ "cgMLST 499616518",
                                           cgmlst_ST == 639048634 ~ "cgMLST 639048634",
                                           cgmlst_ST == 766339150 ~ "cgMLST 766339150",
                                           cgmlst_ST == 778459481 ~ "cgMLST 778459481",
                                           cgmlst_ST == 897612771 ~ "cgMLST 897612771",
                                           cgmlst_ST == 965619992 ~ "cgMLST 965619992",
                                           cgmlst_ST == 983912280 ~ "cgMLST 983912280",
                                           cgmlst_ST == 1021957301 ~ "cgMLST 1021957301",
                                           cgmlst_ST == 1022213880 ~ "cgMLST 1022213880",
                                           cgmlst_ST == 1423174026 ~ "cgMLST 1423174026",
                                           cgmlst_ST == 2187054599 ~ "cgMLST 2187054599",
                                           cgmlst_ST == 3030701707 ~ "cgMLST 3030701707",
                                           cgmlst_ST == 3344264751 ~ "cgMLST 3344264751",
                                           cgmlst_ST == 3917794515 ~ "cgMLST 3917794515",
                                           cgmlst_ST == 4285211217 ~ "cgMLST 4285211217",
                                           cgmlst_ST == 0 ~ "Unclassified"))
d3 <- d3 %>% select(id, cgmlst)

# merge datasets

d4 <- left_join(d1, d2, on = "id")
d5 <- left_join(d4, d3, on = "id")

# enter host data

d6 <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/unmc_genomes/host.csv')

# merge data

d7 <- left_join(d5, d6, on = "id")
d7 <- d7 %>% mutate(baps_1 = as.factor(baps_1))

########################################################################
########################################################################

# Enter Rtab file and select genes of interest

# enter Rtab file and select genes needed

d1b <- read_tsv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/unmc_genomes/gene_presence_absence.Rtab')

# set Genes as rownames

d1b <- column_to_rownames(d1b, var = "Gene") 

d1b$Gene <- NULL

# Transpose it

d2b <- transpose(d1b)

# Add column and rownames

rownames(d2b) <- colnames(d1b)
colnames(d2b) <- rownames(d1b)

# Reset id to colnames

d3b <- rownames_to_column(d2b, "id")

# List of targeted genes 

# metal resistance - SGI3-4

metals <- c("pcoA", "pcoB", "pcoC", "pcoD", "pcoE", "pcoG", "pcoR", "pcoS", "pco",
            "silA", "silB", "silC", "silE", "silP", "silR", "silS", "sil",
            "cusA", "cusB", "cusC", "cusF", "cusR", "cusS", "cus",
            "arsA", "arsB", "arsC", "arsD", "arsR", "ars",
            "merA", "merC", "merP", "merR", "merT",
            "zitB", 
            "zntA", "zntB", "zntR")

######################################################################

d4b <- d3b %>% select("id", contains(metals))

remove_genes <- c("pcoE_2", "cusC_2", "silE_2", "silE_3", "cusS", "cusS_2", "dcuS", "arsA_2", "arsD_2")
d5b <- d4b %>% select(!remove_genes)

# merge with d20

d8 <- left_join(d7, d5b, on = "id")
lociPopStr_st34_sample <- column_to_rownames(d8, var = "id")
write.csv(lociPopStr_st34_sample, "~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/unmc_genomes/lociPopStr_st34_sample.csv")

#####################################################################################
#####################################################################################
#####################################################################################

# enter the phenotype data

phe1 <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/typhimurium/unmc_genomes/in_vitro_ice_typhimurium.csv')

# Change variables

phe1$oxygen[ phe1$oxygen == "aerobic" ] <- "Aerobic"
phe1$oxygen[ phe1$oxygen == "anaerobic" ] <- "Anaerobic"
phe1$ice <- as.factor(phe1$ice)
phe1$copper_concentration <- as.factor(phe1$copper_concentration)
phe1 <- rename(phe1, concentration = copper_concentration)


# plot the data for copper

phe2 <- phe1 %>% filter(heavy_metal == "copper") %>% drop_na()
fig9c <- ggplot(phe2, aes(x = concentration, y = od_values, color = st)) +
    ylim(-0.1, 1) +
    theme_bw() +
    xlab("Copper concentration (mM)") +
    ylab("OD values") +
    theme(axis.text.y = element_text(size = 25)) +
    theme(axis.title.y = element_text(size = 40, face = "bold")) +
    theme(axis.title.x = element_text(size = 40, face = "bold")) +
    theme(axis.text.x = element_text(size = 25, angle = 0, vjust = 1, hjust=0.5)) +
    theme(strip.text.x = element_text(size = 35, colour = "black", angle = 0)) +
    theme(plot.title = element_text(size = 50, face = "bold")) + 
    theme(legend.title=element_text(size=40, face = "bold")) +
    theme(legend.text=element_text(size=25)) +
    geom_boxplot(aes(color = st)) +
    ggtitle("") +  
    labs(color = "ST") +
    facet_wrap(pop_size ~ oxygen) +
    scale_color_manual(values = c("darkblue", "red", "black", "darkgreen")) 
  #geom_smooth(aes(group = st, color = st), method = "loess", se = FALSE, linetype = "dashed", size = 0.5)
fig9c

# statistical model for copper 

model <- aov(od_values ~ pop_size*oxygen*ice, data = phe2)
summary(model)

# Permanova 

#permodel <- adonis(od_values ~ pop_size*oxygen*ice, data = phe2)
#permodel

############################################################################
############################################################################
############################################################################

# plot the data for copper

phe3 <- phe1 %>% filter(heavy_metal == "zinc") %>% drop_na()
fig9d <- ggplot(phe3, aes(x = concentration, y = od_values, color = st)) +
    ylim(-0.1, 1) +
    theme_bw() +
    xlab("Zinc concentration (mM)") +
    ylab("OD values") +
    theme(axis.text.y = element_text(size = 25)) +
    theme(axis.title.y = element_text(size = 40, face = "bold")) +
    theme(axis.title.x = element_text(size = 40, face = "bold")) +
    theme(axis.text.x = element_text(size = 25, angle = 0, vjust = 1, hjust=0.5)) +
    theme(strip.text.x = element_text(size = 35, colour = "black", angle = 0)) +
    theme(plot.title = element_text(size = 50, face = "bold")) + 
    theme(legend.title=element_text(size=40, face = "bold")) +
    theme(legend.text=element_text(size=25)) +
    geom_boxplot(aes(color = st)) +
    ggtitle("") +  
    labs(color = "ST") +
    facet_wrap(pop_size ~ oxygen) +
    scale_color_manual(values = c("darkblue", "red", "black", "darkgreen")) 
  #geom_smooth(aes(group = st, color = st), method = "loess", se = FALSE, linetype = "dashed", size = 0.5)
fig9d

# statistical model for copper 

model2 <- aov(od_values ~ pop_size*oxygen*ice, data = phe3)
summary(model2)
```