---
title: "Infantis cgMLST analysis"
author: "Joao Carlos Gomes-Neto"
date: "8/6/2021"
output: html_document
---

Load packages

```{r, include = FALSE}
# Load packages

library(tidyverse)
library(skimr)
library(RColorBrewer)
library(gridExtra)
library(lattice)
library(ggpubr)
library(vegan)
library(reshape2)
library(ggrepel)
library(ggnewscale)
library(forcats)
library(naniar)
library(data.table)
library(rARPACK)
library(logisticPCA)
library(cluster)
library(factoextra)
library(NbClust)
library(plotly)
library(visdat)
library(ggtree)
library(Rtsne)
library(ggridges)
```

Plot the distribution of new cgMLST variants across Ecotypes and over-time across hosts 

```{r, include = FALSE}
# enter metadata

d1 <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/input_data/metadata_cleaned_final_new.csv')

# rename variables

d2 <- d1 %>% select(id, collection_date, source)

# Combine all years less than 2015 into one group

d2 <- d2 %>%  mutate_all(na_if, "missing")

# change NA in the year column to zero

d2$collection_date[is.na(d2$collection_date)] <- 0

# transform year to numeric

d2 <- d2 %>% mutate(collection_date = as.numeric(collection_date))

# create a new column called Year

d2 <- mutate(d2, year = ifelse(collection_date == 2016, "2016",
                                    ifelse(collection_date == 2017, "2017",
                                            ifelse(collection_date == 2018, "2018",
                                                      ifelse(collection_date == 2019, "2019",
                                                             "<= 2015"
                                           )))))

# check for missing values 

vis_miss(d2)

# filter only needed columns 

d3b <- d2 %>% select(id, year, source)
#############################################################################################

# CGMLST

# Enter SISTR results

sistr <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/input_data/sistr_versions_cgmlst_infantis.csv')

# QC the data 

skim(sistr)

# filter new version of SISTR with new blast version as well

data1 <- sistr %>% filter(variable == "SISTR v1.1 + BLAST v 2.12")

# select columns 

data2 <- data1 %>% select(genome, value)

# rename column 

data2 <- data2 %>% rename(id = genome, cgmlst = value)

# check the data

vis_miss(data2)

# change cgmlst names 

data2 <- mutate(data2, CGMLST = ifelse(cgmlst %in% 1206527699, "cgMLST 1206527699",
                                      ifelse(cgmlst %in% 1000714926, "cgMLST 1000714926",
                                      ifelse(cgmlst %in% 1186491945, "cgMLST 1186491945",
                                      ifelse(cgmlst %in% 989330004, "cgMLST 989330004",
                                            "Other cgMLSTs")))))

# checking for missing values 

sum(is.na(data2))
#############################################################################################
#############################################################################################

# PCA CLUSTER

data3 <- read_csv("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/pca_infantis/cluster2.csv")

data4 <- data3 %>% select(-X1)
#############################################################################################
#############################################################################################

# combining all datasets

data5 <- left_join(data4, data2, on = "id")
data5 <- data5 %>% rename(pca = cluster)

# create a new column

data5 <- data5 %>% mutate(pca_cluster = ifelse(pca == "1", "Cluster 1", "Cluster 2"))
data5 <- data5 %>% select(-pca)
data6 <- left_join(data5, d3b, on = "id")

#######################################################################################
# plot the distribution of cgMLST variants across Ecotypes 
# merge data

data7 <- data6 %>%
              drop_na(pca_cluster, CGMLST) %>%
              select(pca_cluster, CGMLST) %>%
              group_by(pca_cluster, CGMLST) %>%
              summarise(n = n()) %>%
              mutate(total = sum(n)) %>% 
              mutate(prop = n/total*100) 

data7$CGMLST <- factor(data7$CGMLST, levels=c("cgMLST 1206527699", "cgMLST 1000714926", 
                                              "cgMLST 1186491945", "cgMLST 989330004",
                                              "Other cgMLSTs"))

# plot data

plot1 <- ggplot(data7, aes(x = prop, y = pca_cluster, fill = CGMLST)) + xlim(0, 101) +
  theme_bw() +  xlab("Proportion") +
  ylab("") +
    theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 35)) +
  theme(axis.title.x = element_text(size = 40, face = "bold")) +
  theme(axis.text.y = element_text(angle = 0, hjust = 1, size = 40)) +
  theme(strip.text.x = element_text(size = 30, face = "bold")) +
  theme(legend.title=element_text(size=35, face = "bold"))+
  theme(legend.text=element_text(size=30)) +
  scale_fill_manual(name = "cgMLST variants", values=c("gray", "darkred", "orange", "cyan", 
  "darkblue")) +
    geom_bar(position="stack", stat="identity")
plot1

###################################################################################
# plot the distribution of cgMLST variants across hosts over-time 

data8 <- data6 %>%
              select(source, year, CGMLST) %>%
              drop_na(source, year, CGMLST) %>%
              group_by(source, year, CGMLST) %>%
              summarise(n = n()) %>%
              mutate(total = sum(n)) %>% 
              mutate(prop = n/total*100) 
data8$CGMLST <- factor(data8$CGMLST, levels=c("cgMLST 1206527699", "cgMLST 1000714926", 
                                              "cgMLST 1186491945", "cgMLST 989330004",
                                              "Other cgMLSTs"))
plot2 <- ggplot(data8, aes(x = year, y = prop, color = CGMLST, group = CGMLST)) + ylim(0, 100) +
  theme_bw() +  xlab("Years") +
  ylab("Proportion") +
    theme(axis.text.x = element_text(angle = 80, hjust = 1, size = 20)) +
  theme(axis.title.x = element_text(size = 35, face = "bold")) +
  theme(axis.title.y = element_text(size = 35, face = "bold")) +
  theme(axis.text.y = element_text(angle = 0, hjust = 1, size = 30)) +
  theme(legend.title=element_text(size=35, face = "bold"))+
  theme(legend.text=element_text(size=30)) +
  scale_color_manual(name = "cgMLST variants", values=c("gray", "darkred", "orange", "cyan", 
  "darkblue")) +
  theme(strip.text.x = element_text(size = 20, face = "bold")) +
  #guides(linetype = guide_legend(override.aes = list(size = 10)))  +
    geom_line(size=3) +
  theme(legend.key.width = unit(3,"cm")) +
  facet_wrap(~source) 
plot2

data10 <- data6
data10 <- mutate(data10, cgmlst = ifelse(CGMLST %in% 1206527699, "cgMLST 1206527699",
                                      ifelse(CGMLST %in% 1000714926, "cgMLST 1000714926",
                                            "Other cgMLSTs")))
data11 <- data10 %>% select(id, cgmlst)
cgmlst_calls_new_sistr_infantis <- data11
write.csv(cgmlst_calls_new_sistr_infantis, "~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/input_data/cgmlst_calls_new_sistr_infantis.csv")
```

Core-genome distribution of SNPs across cgMLST variants

```{r, include = TRUE}
# CGMLST

# Enter SISTR results

sistr <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/input_data/sistr_versions_cgmlst_infantis.csv')

# QC the data 

skim(sistr)

# filter new version of SISTR with new blast version as well

data1 <- sistr %>% filter(variable == "SISTR v1.1 + BLAST v 2.12")

# select columns 

data2 <- data1 %>% select(genome, value)

# rename column 

data2 <- data2 %>% rename(id = genome, CGMLST = value)

# check the data

vis_miss(data2)

# change cgmlst names 

data2 <- mutate(data2, cgmlst = ifelse(CGMLST %in% 1206527699, "cgMLST 1206527699",
                                      ifelse(CGMLST %in% 1000714926, "cgMLST 1000714926",
                                            "Other cgMLSTs")))

# checking for missing values 

sum(is.na(data2))

# selecting newly classified cgMLST variants

data3 <- data2 %>% select(id, cgmlst)

###############################################################################

# enter the snp data

data4 <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/input_data/distance_snp_sites_infantis_columns.csv')

# filter out distance between itself

data5 <- data4 %>% filter_(~id_1 != id_2)

# distribution of snps 

summary(data5$dist)

# select id_1

d1 <- data5 %>% select(-id_2) %>% rename(id = id_1)
d2 <- left_join(d1, data3, on = "id")
d3 <- d2 %>% rename(id_1 = id)
d3 <- d3 %>% rename(cgmlst_1 = cgmlst) %>% select(-dist)

# select id_2

d1b <- data5 %>% select(-id_1) %>% rename(id = id_2)
d2b <- left_join(d1b, data3, on = "id")
d3b <- d2b %>% rename(id_2 = id)
d3b <- d3b %>% rename(cgmlst_2 = cgmlst)

# merge datasets 

d4 <- bind_cols(d3, d3b)

# remove NA

d5 <- d4 %>% drop_na()

#####################################################################

# table for SNP results 

# cgMLST 1206527699 vs cgMLST 1206527699

d6 <- d5 %>% filter(cgmlst_1 == "cgMLST 1206527699" & cgmlst_2 == "cgMLST 1206527699")
d7 <- d6 %>% select(cgmlst_1, dist) %>% rename(cgmlst = cgmlst_1)

a1 <- d7 %>% 
           summarize(mu = round(mean(dist)), sdx = round(sd(dist)), median = round(median(dist)), rangex = round(range(dist)), iqrx = round(IQR(dist)))
a1
####################################
# cgMLST 1000714926 vs cgMLST 1000714926

d6b <- d5 %>% filter(cgmlst_1 == "cgMLST 1000714926" & cgmlst_2 == "cgMLST 1000714926")
d7b <- d6b %>% select(cgmlst_1, dist) %>% rename(cgmlst = cgmlst_1)

a1b <- d7b %>% 
           summarize(mu = round(mean(dist)), sdx = round(sd(dist)), median = round(median(dist)), rangex = round(range(dist)), iqrx = round(IQR(dist)))
a1b
###########################################
# Other cgMLSTs vs Other cgMLSTs

d6c <- d5 %>% filter(cgmlst_1 == "Other cgMLSTs" & cgmlst_2 == "Other cgMLSTs")
d7c <- d6c %>% select(cgmlst_1, dist) %>% rename(cgmlst = cgmlst_1)

a1c <- d7c %>% 
           summarize(mu = round(mean(dist)), sdx = round(sd(dist)), median = round(median(dist)), rangex = round(range(dist)), iqrx = round(IQR(dist)))
a1c
###########################################

# cgMLST 1206527699 vs cgMLST 1000714926

d6d <- d5 %>% filter(cgmlst_1 == "cgMLST 1206527699" & cgmlst_2 == "cgMLST 1000714926")
d7d <- d6d %>% select(cgmlst_1, dist)
d7d <- d7d %>% mutate(cgmlst = "cgMLST_1206527699_1000714926")
d7d <- d7d %>% select(cgmlst, dist)

a1d <- d7d %>% 
           summarize(mu = round(mean(dist)), sdx = round(sd(dist)), median = round(median(dist)), rangex = round(range(dist)), iqrx = round(IQR(dist)))
a1d
###########################################

# cgMLST 1206527699 vs Other cgMLSTs

d6e <- d5 %>% filter(cgmlst_1 == "cgMLST 1206527699" & cgmlst_2 == "Other cgMLSTs")
d7e <- d6e %>% select(cgmlst_1, dist)
d7e <- d7e %>% mutate(cgmlst = "cgMLST_1206527699_Other_cgMSLTs")
d7e <- d7e %>% select(cgmlst, dist)

a1e <- d7e %>% 
           summarize(mu = round(mean(dist)), sdx = round(sd(dist)), median = round(median(dist)), rangex = round(range(dist)), iqrx = round(IQR(dist)))
a1e
###########################################

# cgMLST 1000714926 vs Other cgMLSTs

d6f <- d5 %>% filter(cgmlst_1 == "cgMLST 1000714926" & cgmlst_2 == "Other cgMLSTs")
d7f <- d6f %>% select(cgmlst_1, dist)
d7f <- d7f %>% mutate(cgmlst = "cgMLST_1000714926_Other_cgMSLTs")
d7f <- d7f %>% select(cgmlst, dist)

a1f <- d7f %>% 
           summarize(mu = round(mean(dist)), sdx = round(sd(dist)), median = round(median(dist)), rangex = round(range(dist)), iqrx = round(IQR(dist)))
a1f

#########################################
# rowbind

d8 <- bind_rows(d7, d7b, d7c, d7d, d7e, d7f)

# Use semi-transparent fill

plot3 <- ggplot(d8, aes(x = dist, fill = cgmlst)) +
  geom_histogram(bins = 30) +
  theme_bw() + 
  theme(legend.position = "none") +
  xlab("Number of SNPs") +
  ylab("Frequency") +
  theme(axis.text.y = element_text(size = 30)) +
  theme(axis.title.y = element_text(size = 35, face = "bold")) +
  theme(axis.title.x = element_text(size = 35, face = "bold")) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 30)) +
  theme(strip.text.x = element_text(size = 18)) +
  scale_fill_manual(values=c("darkblue", "darkgreen", "coral", "black", "cyan", "orange")) +
  facet_wrap(~cgmlst)
plot3
```

Distribution of shell-loci across cgMLST variants 

```{r, include = TRUE}
# CGMLST

# Enter SISTR results

sistr <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/input_data/sistr_versions_cgmlst_infantis.csv')

# QC the data 

skim(sistr)

# filter new version of SISTR with new blast version as well

data1 <- sistr %>% filter(variable == "SISTR v1.1 + BLAST v 2.12")

# select columns 

data2 <- data1 %>% select(genome, value)

# rename column 

data2 <- data2 %>% rename(id = genome, CGMLST = value)

# check the data

vis_miss(data2)

# change cgmlst names 

data2 <- mutate(data2, cgmlst = ifelse(CGMLST %in% 1206527699, "cgMLST 1206527699",
                                      ifelse(CGMLST %in% 1000714926, "cgMLST 1000714926",
                                            "Other cgMLSTs")))

# checking for missing values 

sum(is.na(data2))

# selecting newly classified cgMLST variants

data3 <- data2 %>% select(id, cgmlst)

##############################################################################

# PCA cluster

data4 <- read_csv("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/pca_infantis/cluster2.csv")

d8 <- data4 %>% select(-X1)
d9 <- d8 %>% mutate(pca_cluster = ifelse(cluster == "1", "Cluster 1", "Cluster 2"))
d10 <- d9 %>% select(id, pca_cluster)


# merge datasets

d11 <- left_join(d10, data3, on = "id")

# fill NAs with Other cgMLSTs

d11$cgmlst[is.na(d11$cgmlst)] <- "Other cgMLSTs"

#######################################################################################
#######################################################################################
# enter Rtab file and select genes needed

d1 <- read_tsv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/input_data/gene_presence_absence.Rtab')

# set Genes as rownames

d1 <- column_to_rownames(d1, var = "Gene") 

d1$Gene <- NULL

# Transpose it

d2 <- transpose(d1)

# Add column and rownames

rownames(d2) <- colnames(d1)
colnames(d2) <- rownames(d1)

# Sum up each row

d3 <- d2 %>%
  colSums()

d3b <- as.data.frame(d3)
d3b <- rownames_to_column(d3b, "genes")

# Find genes that are not core (99% or more of genomes)

d4 <- d3b %>% 
      mutate(prop = (d3/2851)*100) %>%
      filter(prop >= 15 & prop < 82)

list_of_genes <- d4$genes

# get the accessory table

d5 <- d2 %>%
      select(list_of_genes)

# Reset id to colnames

d6 <- d5
d6 <- rownames_to_column(d6, "id")

# merge datasets

d13 <- left_join(d11, d6, on = "id")
d13b <- select(d13, -starts_with("group_"))
genes_to_exclude <- c("rhsC", "yicJ_2", "lpfB_2", "ygcS", "nirC", "rluE", "yigB",
                      "wzxC", "sspH2", "hyaB_2", "yicJ_1", "oadG_1", "iolB", 
                      "ompN_2", "uvrA", "mhpE_2", "ushA", "gsk", "oadB_3", "yfdG", "malE",
                      "hybO", "iolG_2", "mhpE_1", "oadB_2", "papC_2", "algC_2",
                      "nanM", "tufA_2", "napA", "tufA_1", "manC1_2", "yjhB_2", "ccmA", "ccmB",
                      "cfiA", "fosA")

d13c <- d13b %>% select(-starts_with(genes_to_exclude))

# export  the data

d14 <- column_to_rownames(d13c, var = "id")
infantis_cgmlst_all_loci <- d14
write.csv(infantis_cgmlst_all_loci, "~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/input_data/infantis_cgmlst_all_loci.csv")
```
Logistic PCA with new cgMLST classification

```{r, include = TRUE}
# CGMLST

# Enter SISTR results

sistr <- read_csv('~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/input_data/sistr_versions_cgmlst_infantis.csv')

# QC the data 

skim(sistr)

# filter new version of SISTR with new blast version as well

data1 <- sistr %>% filter(variable == "SISTR v1.1 + BLAST v 2.12")

# select columns 

data2 <- data1 %>% select(genome, value)

# rename column 

data2 <- data2 %>% rename(id = genome, CGMLST = value)

# check the data

vis_miss(data2)

# change cgmlst names 

data2 <- mutate(data2, cgmlst = ifelse(CGMLST %in% 1206527699, "cgMLST 1206527699",
                                      ifelse(CGMLST %in% 1000714926, "cgMLST 1000714926",
                                            "Other cgMLSTs")))

# checking for missing values 

sum(is.na(data2))

# selecting newly classified cgMLST variants

data3 <- data2 %>% select(id, cgmlst)

#########################################################################################
# enter logistic PCA data

data4 <- read_csv("~/Documents/frontiers_paper_salmonella_newport_typhimurium/data/infantis/pca_infantis/pca_data2.csv")

data5 <- data4 %>% select(-X1)

# merge datasets 

data6 <- left_join(data5, data3, on = "id")
data6$cgmlst[is.na(data6$cgmlst)] <- "Other cgMLSTs"

# plot cgmlst onto PCA

data6$cgmlst <- factor(data6$cgmlst, levels=c("cgMLST 1206527699", "cgMLST 1000714926", "Other cgMLSTs"))
p1 <- ggplot(data6, aes(x = PC1, y = PC2, color = cgmlst)) + 
  theme_bw() + 
  xlab("PC1") +
  ylab("PC2") +
  theme(axis.text.y = element_text(size = 40)) +
  theme(axis.title.y = element_text(size = 45, face = "bold")) +
  theme(axis.title.x = element_text(size = 45, face = "bold")) +
  theme(axis.text.x = element_text(angle = 0, hjust = 1, size = 40)) +
  scale_color_manual(values=c("orange", "black", "gray")) +
  labs(color='cgMLST variants') + 
  geom_jitter(pch = 19, height = 1, width = 1, size = 4) +
  theme(legend.title=element_text(size=50, face = "bold")) +
  theme(legend.text=element_text(size=40)) +
  guides(color = guide_legend(override.aes = list(size=20)))
p1

```
